{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Compare uncorrected, Bonferroni-corrected, and Metasoft eQTL datasets"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": "false",
    "jupyter": {
     "outputs_hidden": "false"
    }
   },
   "source": [
    "## Housekeeping"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": "false",
    "jupyter": {
     "outputs_hidden": "false"
    }
   },
   "source": [
    "### Background\n",
    "\n",
    "* Want to compare the effect of using Bonferroni-correction on number of tissues affected per eQTL and see how this also compares to Metasoft multi-tissue method"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": "false",
    "jupyter": {
     "outputs_hidden": "false"
    }
   },
   "source": [
    "### Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import mysql.connector\n",
    "from sqlalchemy import create_engine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "engine = create_engine('mysql+mysqlconnector://jupyter:password@localhost:3306/gtex', echo=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "def removeGeneIDVersions(text):\n",
    "    return re.findall('(ENSG\\d+)', text)[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": "false",
    "jupyter": {
     "outputs_hidden": "false"
    }
   },
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": "false",
    "jupyter": {
     "outputs_hidden": "false"
    }
   },
   "source": [
    "## Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": "false",
    "jupyter": {
     "outputs_hidden": "false"
    }
   },
   "source": [
    "### Export list of SNP/gene pairs with number of tissues affected"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": "false",
    "jupyter": {
     "outputs_hidden": "false"
    }
   },
   "source": [
    "#### GTEx per tissue data without Bonferroni-correction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "eQTLsAndNumTissuesAffectedUncorrected = pd.read_sql_query(\n",
    "    'SELECT variant_id, gene_id, count(*) AS `count` FROM `v7` GROUP BY variant_id, gene_id',\n",
    "    engine,\n",
    "    coerce_float=True\n",
    ")\n",
    "eQTLsAndNumTissuesAffectedUncorrected['gene_id'] = eQTLsAndNumTissuesAffectedUncorrected['gene_id'].apply(removeGeneIDVersions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>7.627598e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>4.822141e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>7.782665e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>2.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>4.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>4.800000e+01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              count\n",
       "count  7.627598e+06\n",
       "mean   4.822141e+00\n",
       "std    7.782665e+00\n",
       "min    1.000000e+00\n",
       "25%    1.000000e+00\n",
       "50%    2.000000e+00\n",
       "75%    4.000000e+00\n",
       "max    4.800000e+01"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eQTLsAndNumTissuesAffectedUncorrected.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": "false",
    "jupyter": {
     "outputs_hidden": "false"
    }
   },
   "source": [
    "#### Bonferroni-corrected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "eQTLsAndNumTissuesAffectedBonferroni = pd.read_sql_query(\n",
    "    'SELECT variant_id, gene_id, count(*) AS `count` FROM `v7` WHERE sigAfterBonferroni = 1 GROUP BY variant_id, gene_id',\n",
    "    engine,\n",
    "    coerce_float=True\n",
    ")\n",
    "eQTLsAndNumTissuesAffectedBonferroni['gene_id'] = eQTLsAndNumTissuesAffectedBonferroni['gene_id'].apply(removeGeneIDVersions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>4.487596e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>5.061874e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>8.082678e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>2.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>5.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>4.800000e+01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              count\n",
       "count  4.487596e+06\n",
       "mean   5.061874e+00\n",
       "std    8.082678e+00\n",
       "min    1.000000e+00\n",
       "25%    1.000000e+00\n",
       "50%    2.000000e+00\n",
       "75%    5.000000e+00\n",
       "max    4.800000e+01"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eQTLsAndNumTissuesAffectedBonferroni.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": "false",
    "jupyter": {
     "outputs_hidden": "false"
    }
   },
   "source": [
    "#### Metasoft eQTLs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "eQTLsAndNumTissuesAffectedMetasoft = pd.read_sql_query(\n",
    "    'SELECT snp AS `variant_id`, gene AS gene_id, mvalTissues AS `count` FROM `v7Metasoft`',\n",
    "    engine,\n",
    "    coerce_float=True\n",
    ")\n",
    "eQTLsAndNumTissuesAffectedMetasoft['gene_id'] = eQTLsAndNumTissuesAffectedMetasoft['gene_id'].apply(removeGeneIDVersions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>7.627598e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>1.651631e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>1.561892e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>2.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>1.100000e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>3.000000e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>4.800000e+01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              count\n",
       "count  7.627598e+06\n",
       "mean   1.651631e+01\n",
       "std    1.561892e+01\n",
       "min    0.000000e+00\n",
       "25%    2.000000e+00\n",
       "50%    1.100000e+01\n",
       "75%    3.000000e+01\n",
       "max    4.800000e+01"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eQTLsAndNumTissuesAffectedMetasoft.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": "false",
    "jupyter": {
     "outputs_hidden": "false"
    }
   },
   "source": [
    "#### Merge number of tissues affected for the three datasets for same SNP/gene pairs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "eQTLsAndNumTissuesAffectedUncorrected.rename(columns={'count':'countUncorrected'}, inplace=True)\n",
    "eQTLsAndNumTissuesAffectedBonferroni.rename(columns={'count':'countBonferroniCorrected'}, inplace=True)\n",
    "eQTLsAndNumTissuesAffectedMetasoft.rename(columns={'count':'countMetasoft'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "merge1 = pd.merge(eQTLsAndNumTissuesAffectedUncorrected,eQTLsAndNumTissuesAffectedBonferroni, how='left', on=['variant_id','gene_id'])\n",
    "eQTLsAndNumTissuesAffected = pd.merge(merge1,eQTLsAndNumTissuesAffectedMetasoft, how='left', on=['variant_id','gene_id'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": "false",
    "jupyter": {
     "outputs_hidden": "false"
    }
   },
   "source": [
    "Where no tissue affected for Bonferroni-corrected data, replace NA with 0."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "eQTLsAndNumTissuesAffected.fillna(0, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>countUncorrected</th>\n",
       "      <th>countBonferroniCorrected</th>\n",
       "      <th>countMetasoft</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10_100000625_A_G_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>5</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10_100000625_A_G_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10_100000625_A_G_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>4</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10_100000645_A_C_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10_100000645_A_C_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>11</td>\n",
       "      <td>10.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10_100002841_C_CT_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>10_100003242_T_G_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>10_100003242_T_G_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10_100003785_T_C_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10_100003785_T_C_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10_100003785_T_C_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>10_100004360_G_A_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>10_100004360_G_A_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>11</td>\n",
       "      <td>10.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>10_100004441_G_C_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>10_100004441_G_C_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>10_100004441_G_C_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>10_100004906_C_A_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>5</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>10_100004906_C_A_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>10_100004906_C_A_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>10_100004996_G_A_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>10_100004996_G_A_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>10_100004996_G_A_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>10_100005282_C_T_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>5</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>10_100005282_C_T_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>10_100005282_C_T_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>10_100007362_G_C_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>10_100007362_G_C_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>11</td>\n",
       "      <td>10.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>10_100008436_G_A_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>10_100008436_G_A_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>10_100008436_G_A_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627568</th>\n",
       "      <td>X_99994917_G_A_b37</td>\n",
       "      <td>ENSG00000101811</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627569</th>\n",
       "      <td>X_99994917_G_A_b37</td>\n",
       "      <td>ENSG00000102362</td>\n",
       "      <td>21</td>\n",
       "      <td>20.0</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627570</th>\n",
       "      <td>X_99994917_G_A_b37</td>\n",
       "      <td>ENSG00000223819</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627571</th>\n",
       "      <td>X_99995091_A_G_b37</td>\n",
       "      <td>ENSG00000261101</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627572</th>\n",
       "      <td>X_99995817_T_C_b37</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627573</th>\n",
       "      <td>X_99995817_T_C_b37</td>\n",
       "      <td>ENSG00000007952</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627574</th>\n",
       "      <td>X_99995817_T_C_b37</td>\n",
       "      <td>ENSG00000101811</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627575</th>\n",
       "      <td>X_99995817_T_C_b37</td>\n",
       "      <td>ENSG00000102362</td>\n",
       "      <td>24</td>\n",
       "      <td>21.0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627576</th>\n",
       "      <td>X_99995817_T_C_b37</td>\n",
       "      <td>ENSG00000165194</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627577</th>\n",
       "      <td>X_99995817_T_C_b37</td>\n",
       "      <td>ENSG00000223819</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627578</th>\n",
       "      <td>X_99995933_C_G_b37</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627579</th>\n",
       "      <td>X_99995933_C_G_b37</td>\n",
       "      <td>ENSG00000101811</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627580</th>\n",
       "      <td>X_99995933_C_G_b37</td>\n",
       "      <td>ENSG00000102362</td>\n",
       "      <td>26</td>\n",
       "      <td>22.0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627581</th>\n",
       "      <td>X_99995933_C_G_b37</td>\n",
       "      <td>ENSG00000223819</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627582</th>\n",
       "      <td>X_99997406_T_G_b37</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627583</th>\n",
       "      <td>X_99997406_T_G_b37</td>\n",
       "      <td>ENSG00000007952</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627584</th>\n",
       "      <td>X_99997406_T_G_b37</td>\n",
       "      <td>ENSG00000101811</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627585</th>\n",
       "      <td>X_99997406_T_G_b37</td>\n",
       "      <td>ENSG00000102362</td>\n",
       "      <td>26</td>\n",
       "      <td>21.0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627586</th>\n",
       "      <td>X_99997406_T_G_b37</td>\n",
       "      <td>ENSG00000223819</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627587</th>\n",
       "      <td>X_99998829_C_T_b37</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627588</th>\n",
       "      <td>X_99998829_C_T_b37</td>\n",
       "      <td>ENSG00000007952</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627589</th>\n",
       "      <td>X_99998829_C_T_b37</td>\n",
       "      <td>ENSG00000101811</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627590</th>\n",
       "      <td>X_99998829_C_T_b37</td>\n",
       "      <td>ENSG00000102362</td>\n",
       "      <td>25</td>\n",
       "      <td>20.0</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627591</th>\n",
       "      <td>X_99998829_C_T_b37</td>\n",
       "      <td>ENSG00000165194</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627592</th>\n",
       "      <td>X_99998829_C_T_b37</td>\n",
       "      <td>ENSG00000223819</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627593</th>\n",
       "      <td>X_99999349_G_A_b37</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627594</th>\n",
       "      <td>X_99999349_G_A_b37</td>\n",
       "      <td>ENSG00000007952</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627595</th>\n",
       "      <td>X_99999349_G_A_b37</td>\n",
       "      <td>ENSG00000101811</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627596</th>\n",
       "      <td>X_99999349_G_A_b37</td>\n",
       "      <td>ENSG00000102362</td>\n",
       "      <td>24</td>\n",
       "      <td>20.0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627597</th>\n",
       "      <td>X_99999349_G_A_b37</td>\n",
       "      <td>ENSG00000223819</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7627598 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    variant_id          gene_id  countUncorrected  \\\n",
       "0         10_100000625_A_G_b37  ENSG00000138131                 5   \n",
       "1         10_100000625_A_G_b37  ENSG00000166024                 4   \n",
       "2         10_100000625_A_G_b37  ENSG00000230928                 4   \n",
       "3         10_100000645_A_C_b37  ENSG00000138131                 1   \n",
       "4         10_100000645_A_C_b37  ENSG00000230928                11   \n",
       "5        10_100002841_C_CT_b37  ENSG00000230928                 2   \n",
       "6         10_100003242_T_G_b37  ENSG00000166024                 1   \n",
       "7         10_100003242_T_G_b37  ENSG00000230928                 1   \n",
       "8         10_100003785_T_C_b37  ENSG00000138131                 6   \n",
       "9         10_100003785_T_C_b37  ENSG00000166024                 4   \n",
       "10        10_100003785_T_C_b37  ENSG00000230928                 3   \n",
       "11        10_100004360_G_A_b37  ENSG00000138131                 1   \n",
       "12        10_100004360_G_A_b37  ENSG00000230928                11   \n",
       "13        10_100004441_G_C_b37  ENSG00000138131                 5   \n",
       "14        10_100004441_G_C_b37  ENSG00000166024                 4   \n",
       "15        10_100004441_G_C_b37  ENSG00000230928                 4   \n",
       "16        10_100004906_C_A_b37  ENSG00000138131                 5   \n",
       "17        10_100004906_C_A_b37  ENSG00000166024                 4   \n",
       "18        10_100004906_C_A_b37  ENSG00000230928                 5   \n",
       "19        10_100004996_G_A_b37  ENSG00000138131                 6   \n",
       "20        10_100004996_G_A_b37  ENSG00000166024                 4   \n",
       "21        10_100004996_G_A_b37  ENSG00000230928                 3   \n",
       "22        10_100005282_C_T_b37  ENSG00000138131                 5   \n",
       "23        10_100005282_C_T_b37  ENSG00000166024                 4   \n",
       "24        10_100005282_C_T_b37  ENSG00000230928                 5   \n",
       "25        10_100007362_G_C_b37  ENSG00000138131                 1   \n",
       "26        10_100007362_G_C_b37  ENSG00000230928                11   \n",
       "27        10_100008436_G_A_b37  ENSG00000138131                 6   \n",
       "28        10_100008436_G_A_b37  ENSG00000166024                 4   \n",
       "29        10_100008436_G_A_b37  ENSG00000230928                 3   \n",
       "...                        ...              ...               ...   \n",
       "7627568     X_99994917_G_A_b37  ENSG00000101811                 1   \n",
       "7627569     X_99994917_G_A_b37  ENSG00000102362                21   \n",
       "7627570     X_99994917_G_A_b37  ENSG00000223819                 1   \n",
       "7627571     X_99995091_A_G_b37  ENSG00000261101                 1   \n",
       "7627572     X_99995817_T_C_b37  ENSG00000000003                 2   \n",
       "7627573     X_99995817_T_C_b37  ENSG00000007952                 1   \n",
       "7627574     X_99995817_T_C_b37  ENSG00000101811                 1   \n",
       "7627575     X_99995817_T_C_b37  ENSG00000102362                24   \n",
       "7627576     X_99995817_T_C_b37  ENSG00000165194                 1   \n",
       "7627577     X_99995817_T_C_b37  ENSG00000223819                 1   \n",
       "7627578     X_99995933_C_G_b37  ENSG00000000003                 3   \n",
       "7627579     X_99995933_C_G_b37  ENSG00000101811                 1   \n",
       "7627580     X_99995933_C_G_b37  ENSG00000102362                26   \n",
       "7627581     X_99995933_C_G_b37  ENSG00000223819                 1   \n",
       "7627582     X_99997406_T_G_b37  ENSG00000000003                 2   \n",
       "7627583     X_99997406_T_G_b37  ENSG00000007952                 1   \n",
       "7627584     X_99997406_T_G_b37  ENSG00000101811                 1   \n",
       "7627585     X_99997406_T_G_b37  ENSG00000102362                26   \n",
       "7627586     X_99997406_T_G_b37  ENSG00000223819                 1   \n",
       "7627587     X_99998829_C_T_b37  ENSG00000000003                 2   \n",
       "7627588     X_99998829_C_T_b37  ENSG00000007952                 1   \n",
       "7627589     X_99998829_C_T_b37  ENSG00000101811                 1   \n",
       "7627590     X_99998829_C_T_b37  ENSG00000102362                25   \n",
       "7627591     X_99998829_C_T_b37  ENSG00000165194                 1   \n",
       "7627592     X_99998829_C_T_b37  ENSG00000223819                 1   \n",
       "7627593     X_99999349_G_A_b37  ENSG00000000003                 2   \n",
       "7627594     X_99999349_G_A_b37  ENSG00000007952                 1   \n",
       "7627595     X_99999349_G_A_b37  ENSG00000101811                 1   \n",
       "7627596     X_99999349_G_A_b37  ENSG00000102362                24   \n",
       "7627597     X_99999349_G_A_b37  ENSG00000223819                 1   \n",
       "\n",
       "         countBonferroniCorrected  countMetasoft  \n",
       "0                             4.0              5  \n",
       "1                             3.0              9  \n",
       "2                             3.0              9  \n",
       "3                             1.0              2  \n",
       "4                            10.0             29  \n",
       "5                             0.0             14  \n",
       "6                             0.0              9  \n",
       "7                             0.0              4  \n",
       "8                             3.0              5  \n",
       "9                             2.0             12  \n",
       "10                            2.0             10  \n",
       "11                            1.0              2  \n",
       "12                           10.0             29  \n",
       "13                            3.0              5  \n",
       "14                            1.0             10  \n",
       "15                            2.0              9  \n",
       "16                            5.0              6  \n",
       "17                            2.0              9  \n",
       "18                            3.0              9  \n",
       "19                            3.0              5  \n",
       "20                            2.0             13  \n",
       "21                            2.0              9  \n",
       "22                            5.0              5  \n",
       "23                            2.0              9  \n",
       "24                            3.0              9  \n",
       "25                            1.0              2  \n",
       "26                           10.0             29  \n",
       "27                            3.0              5  \n",
       "28                            2.0             13  \n",
       "29                            2.0              9  \n",
       "...                           ...            ...  \n",
       "7627568                       1.0              0  \n",
       "7627569                      20.0             21  \n",
       "7627570                       1.0              0  \n",
       "7627571                       0.0              6  \n",
       "7627572                       1.0             15  \n",
       "7627573                       0.0             29  \n",
       "7627574                       1.0              1  \n",
       "7627575                      21.0             20  \n",
       "7627576                       0.0              3  \n",
       "7627577                       1.0              0  \n",
       "7627578                       1.0             15  \n",
       "7627579                       1.0              1  \n",
       "7627580                      22.0             20  \n",
       "7627581                       1.0              0  \n",
       "7627582                       2.0             19  \n",
       "7627583                       0.0             25  \n",
       "7627584                       1.0              1  \n",
       "7627585                      21.0             20  \n",
       "7627586                       0.0              0  \n",
       "7627587                       2.0             15  \n",
       "7627588                       0.0             23  \n",
       "7627589                       1.0              1  \n",
       "7627590                      20.0             21  \n",
       "7627591                       0.0              1  \n",
       "7627592                       1.0              0  \n",
       "7627593                       2.0             11  \n",
       "7627594                       0.0             26  \n",
       "7627595                       1.0              1  \n",
       "7627596                      20.0             20  \n",
       "7627597                       1.0              0  \n",
       "\n",
       "[7627598 rows x 5 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eQTLsAndNumTissuesAffected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>countUncorrected</th>\n",
       "      <th>countBonferroniCorrected</th>\n",
       "      <th>countMetasoft</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>7.627598e+06</td>\n",
       "      <td>7.627598e+06</td>\n",
       "      <td>7.627598e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>4.822141e+00</td>\n",
       "      <td>2.978086e+00</td>\n",
       "      <td>1.651631e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>7.782665e+00</td>\n",
       "      <td>6.681433e+00</td>\n",
       "      <td>1.561892e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>2.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.100000e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>4.000000e+00</td>\n",
       "      <td>2.000000e+00</td>\n",
       "      <td>3.000000e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>4.800000e+01</td>\n",
       "      <td>4.800000e+01</td>\n",
       "      <td>4.800000e+01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       countUncorrected  countBonferroniCorrected  countMetasoft\n",
       "count      7.627598e+06              7.627598e+06   7.627598e+06\n",
       "mean       4.822141e+00              2.978086e+00   1.651631e+01\n",
       "std        7.782665e+00              6.681433e+00   1.561892e+01\n",
       "min        1.000000e+00              0.000000e+00   0.000000e+00\n",
       "25%        1.000000e+00              0.000000e+00   2.000000e+00\n",
       "50%        2.000000e+00              1.000000e+00   1.100000e+01\n",
       "75%        4.000000e+00              2.000000e+00   3.000000e+01\n",
       "max        4.800000e+01              4.800000e+01   4.800000e+01"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eQTLsAndNumTissuesAffected.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "eQTLsAndNumTissuesAffected.to_csv('../outputFiles/GTExV7/eQTLsAndNumTissuesAffected.txt',\n",
    "                                 index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Calculate the number of protein coding genes that have a significant eQTL in GTEx V7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>countUncorrected</th>\n",
       "      <th>countBonferroniCorrected</th>\n",
       "      <th>countMetasoft</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10_100000625_A_G_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>5</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10_100000625_A_G_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10_100000625_A_G_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>4</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10_100000645_A_C_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10_100000645_A_C_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>11</td>\n",
       "      <td>10.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10_100002841_C_CT_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>10_100003242_T_G_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>10_100003242_T_G_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10_100003785_T_C_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10_100003785_T_C_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10_100003785_T_C_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>10_100004360_G_A_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>10_100004360_G_A_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>11</td>\n",
       "      <td>10.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>10_100004441_G_C_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>10_100004441_G_C_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>10_100004441_G_C_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>10_100004906_C_A_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>5</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>10_100004906_C_A_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>10_100004906_C_A_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>10_100004996_G_A_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>10_100004996_G_A_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>10_100004996_G_A_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>10_100005282_C_T_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>5</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>10_100005282_C_T_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>10_100005282_C_T_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>10_100007362_G_C_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>10_100007362_G_C_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>11</td>\n",
       "      <td>10.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>10_100008436_G_A_b37</td>\n",
       "      <td>ENSG00000138131</td>\n",
       "      <td>6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>10_100008436_G_A_b37</td>\n",
       "      <td>ENSG00000166024</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>10_100008436_G_A_b37</td>\n",
       "      <td>ENSG00000230928</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627568</th>\n",
       "      <td>X_99994917_G_A_b37</td>\n",
       "      <td>ENSG00000101811</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627569</th>\n",
       "      <td>X_99994917_G_A_b37</td>\n",
       "      <td>ENSG00000102362</td>\n",
       "      <td>21</td>\n",
       "      <td>20.0</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627570</th>\n",
       "      <td>X_99994917_G_A_b37</td>\n",
       "      <td>ENSG00000223819</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627571</th>\n",
       "      <td>X_99995091_A_G_b37</td>\n",
       "      <td>ENSG00000261101</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627572</th>\n",
       "      <td>X_99995817_T_C_b37</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627573</th>\n",
       "      <td>X_99995817_T_C_b37</td>\n",
       "      <td>ENSG00000007952</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627574</th>\n",
       "      <td>X_99995817_T_C_b37</td>\n",
       "      <td>ENSG00000101811</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627575</th>\n",
       "      <td>X_99995817_T_C_b37</td>\n",
       "      <td>ENSG00000102362</td>\n",
       "      <td>24</td>\n",
       "      <td>21.0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627576</th>\n",
       "      <td>X_99995817_T_C_b37</td>\n",
       "      <td>ENSG00000165194</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627577</th>\n",
       "      <td>X_99995817_T_C_b37</td>\n",
       "      <td>ENSG00000223819</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627578</th>\n",
       "      <td>X_99995933_C_G_b37</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627579</th>\n",
       "      <td>X_99995933_C_G_b37</td>\n",
       "      <td>ENSG00000101811</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627580</th>\n",
       "      <td>X_99995933_C_G_b37</td>\n",
       "      <td>ENSG00000102362</td>\n",
       "      <td>26</td>\n",
       "      <td>22.0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627581</th>\n",
       "      <td>X_99995933_C_G_b37</td>\n",
       "      <td>ENSG00000223819</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627582</th>\n",
       "      <td>X_99997406_T_G_b37</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627583</th>\n",
       "      <td>X_99997406_T_G_b37</td>\n",
       "      <td>ENSG00000007952</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627584</th>\n",
       "      <td>X_99997406_T_G_b37</td>\n",
       "      <td>ENSG00000101811</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627585</th>\n",
       "      <td>X_99997406_T_G_b37</td>\n",
       "      <td>ENSG00000102362</td>\n",
       "      <td>26</td>\n",
       "      <td>21.0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627586</th>\n",
       "      <td>X_99997406_T_G_b37</td>\n",
       "      <td>ENSG00000223819</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627587</th>\n",
       "      <td>X_99998829_C_T_b37</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627588</th>\n",
       "      <td>X_99998829_C_T_b37</td>\n",
       "      <td>ENSG00000007952</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627589</th>\n",
       "      <td>X_99998829_C_T_b37</td>\n",
       "      <td>ENSG00000101811</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627590</th>\n",
       "      <td>X_99998829_C_T_b37</td>\n",
       "      <td>ENSG00000102362</td>\n",
       "      <td>25</td>\n",
       "      <td>20.0</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627591</th>\n",
       "      <td>X_99998829_C_T_b37</td>\n",
       "      <td>ENSG00000165194</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627592</th>\n",
       "      <td>X_99998829_C_T_b37</td>\n",
       "      <td>ENSG00000223819</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627593</th>\n",
       "      <td>X_99999349_G_A_b37</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627594</th>\n",
       "      <td>X_99999349_G_A_b37</td>\n",
       "      <td>ENSG00000007952</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627595</th>\n",
       "      <td>X_99999349_G_A_b37</td>\n",
       "      <td>ENSG00000101811</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627596</th>\n",
       "      <td>X_99999349_G_A_b37</td>\n",
       "      <td>ENSG00000102362</td>\n",
       "      <td>24</td>\n",
       "      <td>20.0</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7627597</th>\n",
       "      <td>X_99999349_G_A_b37</td>\n",
       "      <td>ENSG00000223819</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7627598 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    variant_id          gene_id  countUncorrected  \\\n",
       "0         10_100000625_A_G_b37  ENSG00000138131                 5   \n",
       "1         10_100000625_A_G_b37  ENSG00000166024                 4   \n",
       "2         10_100000625_A_G_b37  ENSG00000230928                 4   \n",
       "3         10_100000645_A_C_b37  ENSG00000138131                 1   \n",
       "4         10_100000645_A_C_b37  ENSG00000230928                11   \n",
       "5        10_100002841_C_CT_b37  ENSG00000230928                 2   \n",
       "6         10_100003242_T_G_b37  ENSG00000166024                 1   \n",
       "7         10_100003242_T_G_b37  ENSG00000230928                 1   \n",
       "8         10_100003785_T_C_b37  ENSG00000138131                 6   \n",
       "9         10_100003785_T_C_b37  ENSG00000166024                 4   \n",
       "10        10_100003785_T_C_b37  ENSG00000230928                 3   \n",
       "11        10_100004360_G_A_b37  ENSG00000138131                 1   \n",
       "12        10_100004360_G_A_b37  ENSG00000230928                11   \n",
       "13        10_100004441_G_C_b37  ENSG00000138131                 5   \n",
       "14        10_100004441_G_C_b37  ENSG00000166024                 4   \n",
       "15        10_100004441_G_C_b37  ENSG00000230928                 4   \n",
       "16        10_100004906_C_A_b37  ENSG00000138131                 5   \n",
       "17        10_100004906_C_A_b37  ENSG00000166024                 4   \n",
       "18        10_100004906_C_A_b37  ENSG00000230928                 5   \n",
       "19        10_100004996_G_A_b37  ENSG00000138131                 6   \n",
       "20        10_100004996_G_A_b37  ENSG00000166024                 4   \n",
       "21        10_100004996_G_A_b37  ENSG00000230928                 3   \n",
       "22        10_100005282_C_T_b37  ENSG00000138131                 5   \n",
       "23        10_100005282_C_T_b37  ENSG00000166024                 4   \n",
       "24        10_100005282_C_T_b37  ENSG00000230928                 5   \n",
       "25        10_100007362_G_C_b37  ENSG00000138131                 1   \n",
       "26        10_100007362_G_C_b37  ENSG00000230928                11   \n",
       "27        10_100008436_G_A_b37  ENSG00000138131                 6   \n",
       "28        10_100008436_G_A_b37  ENSG00000166024                 4   \n",
       "29        10_100008436_G_A_b37  ENSG00000230928                 3   \n",
       "...                        ...              ...               ...   \n",
       "7627568     X_99994917_G_A_b37  ENSG00000101811                 1   \n",
       "7627569     X_99994917_G_A_b37  ENSG00000102362                21   \n",
       "7627570     X_99994917_G_A_b37  ENSG00000223819                 1   \n",
       "7627571     X_99995091_A_G_b37  ENSG00000261101                 1   \n",
       "7627572     X_99995817_T_C_b37  ENSG00000000003                 2   \n",
       "7627573     X_99995817_T_C_b37  ENSG00000007952                 1   \n",
       "7627574     X_99995817_T_C_b37  ENSG00000101811                 1   \n",
       "7627575     X_99995817_T_C_b37  ENSG00000102362                24   \n",
       "7627576     X_99995817_T_C_b37  ENSG00000165194                 1   \n",
       "7627577     X_99995817_T_C_b37  ENSG00000223819                 1   \n",
       "7627578     X_99995933_C_G_b37  ENSG00000000003                 3   \n",
       "7627579     X_99995933_C_G_b37  ENSG00000101811                 1   \n",
       "7627580     X_99995933_C_G_b37  ENSG00000102362                26   \n",
       "7627581     X_99995933_C_G_b37  ENSG00000223819                 1   \n",
       "7627582     X_99997406_T_G_b37  ENSG00000000003                 2   \n",
       "7627583     X_99997406_T_G_b37  ENSG00000007952                 1   \n",
       "7627584     X_99997406_T_G_b37  ENSG00000101811                 1   \n",
       "7627585     X_99997406_T_G_b37  ENSG00000102362                26   \n",
       "7627586     X_99997406_T_G_b37  ENSG00000223819                 1   \n",
       "7627587     X_99998829_C_T_b37  ENSG00000000003                 2   \n",
       "7627588     X_99998829_C_T_b37  ENSG00000007952                 1   \n",
       "7627589     X_99998829_C_T_b37  ENSG00000101811                 1   \n",
       "7627590     X_99998829_C_T_b37  ENSG00000102362                25   \n",
       "7627591     X_99998829_C_T_b37  ENSG00000165194                 1   \n",
       "7627592     X_99998829_C_T_b37  ENSG00000223819                 1   \n",
       "7627593     X_99999349_G_A_b37  ENSG00000000003                 2   \n",
       "7627594     X_99999349_G_A_b37  ENSG00000007952                 1   \n",
       "7627595     X_99999349_G_A_b37  ENSG00000101811                 1   \n",
       "7627596     X_99999349_G_A_b37  ENSG00000102362                24   \n",
       "7627597     X_99999349_G_A_b37  ENSG00000223819                 1   \n",
       "\n",
       "         countBonferroniCorrected  countMetasoft  \n",
       "0                             4.0              5  \n",
       "1                             3.0              9  \n",
       "2                             3.0              9  \n",
       "3                             1.0              2  \n",
       "4                            10.0             29  \n",
       "5                             0.0             14  \n",
       "6                             0.0              9  \n",
       "7                             0.0              4  \n",
       "8                             3.0              5  \n",
       "9                             2.0             12  \n",
       "10                            2.0             10  \n",
       "11                            1.0              2  \n",
       "12                           10.0             29  \n",
       "13                            3.0              5  \n",
       "14                            1.0             10  \n",
       "15                            2.0              9  \n",
       "16                            5.0              6  \n",
       "17                            2.0              9  \n",
       "18                            3.0              9  \n",
       "19                            3.0              5  \n",
       "20                            2.0             13  \n",
       "21                            2.0              9  \n",
       "22                            5.0              5  \n",
       "23                            2.0              9  \n",
       "24                            3.0              9  \n",
       "25                            1.0              2  \n",
       "26                           10.0             29  \n",
       "27                            3.0              5  \n",
       "28                            2.0             13  \n",
       "29                            2.0              9  \n",
       "...                           ...            ...  \n",
       "7627568                       1.0              0  \n",
       "7627569                      20.0             21  \n",
       "7627570                       1.0              0  \n",
       "7627571                       0.0              6  \n",
       "7627572                       1.0             15  \n",
       "7627573                       0.0             29  \n",
       "7627574                       1.0              1  \n",
       "7627575                      21.0             20  \n",
       "7627576                       0.0              3  \n",
       "7627577                       1.0              0  \n",
       "7627578                       1.0             15  \n",
       "7627579                       1.0              1  \n",
       "7627580                      22.0             20  \n",
       "7627581                       1.0              0  \n",
       "7627582                       2.0             19  \n",
       "7627583                       0.0             25  \n",
       "7627584                       1.0              1  \n",
       "7627585                      21.0             20  \n",
       "7627586                       0.0              0  \n",
       "7627587                       2.0             15  \n",
       "7627588                       0.0             23  \n",
       "7627589                       1.0              1  \n",
       "7627590                      20.0             21  \n",
       "7627591                       0.0              1  \n",
       "7627592                       1.0              0  \n",
       "7627593                       2.0             11  \n",
       "7627594                       0.0             26  \n",
       "7627595                       1.0              1  \n",
       "7627596                      20.0             20  \n",
       "7627597                       1.0              0  \n",
       "\n",
       "[7627598 rows x 5 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eQTLsAndNumTissuesAffected = pd.read_csv('../outputFiles/GTExV7/eQTLsAndNumTissuesAffected.txt')\n",
    "eQTLsAndNumTissuesAffected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "affectedGenes = eQTLsAndNumTissuesAffected['gene_id'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "31385"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(affectedGenes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ensembl Gene ID</th>\n",
       "      <th>Chromosome Name</th>\n",
       "      <th>Gene Start (bp)</th>\n",
       "      <th>Gene End (bp)</th>\n",
       "      <th>Strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000215405</td>\n",
       "      <td>15</td>\n",
       "      <td>20737094</td>\n",
       "      <td>20747114</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000268343</td>\n",
       "      <td>15</td>\n",
       "      <td>21004687</td>\n",
       "      <td>21005367</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000230031</td>\n",
       "      <td>15</td>\n",
       "      <td>21040701</td>\n",
       "      <td>21071643</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000138593</td>\n",
       "      <td>15</td>\n",
       "      <td>49280673</td>\n",
       "      <td>49338760</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000268531</td>\n",
       "      <td>15</td>\n",
       "      <td>22011370</td>\n",
       "      <td>22012050</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Ensembl Gene ID Chromosome Name  Gene Start (bp)  Gene End (bp)  Strand\n",
       "0  ENSG00000215405              15         20737094       20747114      -1\n",
       "1  ENSG00000268343              15         21004687       21005367       1\n",
       "2  ENSG00000230031              15         21040701       21071643      -1\n",
       "3  ENSG00000138593              15         49280673       49338760      -1\n",
       "4  ENSG00000268531              15         22011370       22012050       1"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PCGenes = pd.read_csv('../datasets/geneLists/Ensembl/EnsV75ProteinCodingGenes1-Y.txt', sep='\\t')\n",
    "PCGenes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "18199"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(PCGenes[PCGenes['Ensembl Gene ID'].isin(affectedGenes)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:paper-eqtls]",
   "language": "python",
   "name": "conda-env-paper-eqtls-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
