{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "dba0c998-c43c-4a87-8404-3a8370571ffd",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1dd28bba-376b-4285-a907-4e78ce427794",
   "metadata": {},
   "outputs": [],
   "source": [
    "tissues = ['Adipose',\n",
    "'Blood',\n",
    "'Embryo',\n",
    "'Hypothalamus',\n",
    "'Intramuscular_fat',\n",
    "'Jejunum',\n",
    "'Kidney',\n",
    "'Leukocyte',\n",
    "'Liver',\n",
    "'Lung',\n",
    "'Lymph_node',\n",
    "'Macrophage',\n",
    "'Mammary',\n",
    "#'Mammary_L',\n",
    "'Milk_cell',\n",
    "'Monocytes',\n",
    "'Muscle',\n",
    "#'Muscle_cross',\n",
    "#'Muscle_indicus',\n",
    "#'Muscle_taurus',\n",
    "'Ovary',\n",
    "'Oviduct',\n",
    "'Pituitary',\n",
    "'Rumen',\n",
    "'Spleen',\n",
    "'Testis',\n",
    "'Uterus']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "054b47b8-6c9e-46d9-897a-b89f19df66d7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Adipose.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18057 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00279622 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Blood.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 16179 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9999 \n",
      "  * Corrected p-value threshold =  0.0221577 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Embryo.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 14097 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00559607 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Hypothalamus.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18588 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.001695535 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Intramuscular_fat.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 14928 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00741407 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Jejunum.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 17995 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00159072 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Kidney.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18967 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9996 \n",
      "  * Corrected p-value threshold =  0.000207602 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Leukocyte.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 14986 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.0006661515 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Liver.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 16558 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.0132995 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Lung.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 19125 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00488796 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Lymph_node.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18160 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00540915 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Macrophage.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 13753 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.0211331 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Mammary.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18222 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.006192955 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Mammary_L.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 16332 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.0262197 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Milk_cell.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 14881 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00121119 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Monocytes.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 15902 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.006728795 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Muscle.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 15451 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.0135557 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Muscle_cross.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 15497 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00258496 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Muscle_indicus.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 15291 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.000286652 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Muscle_taurus.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 15222 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00908362 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Ovary.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 17526 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00177968 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Oviduct.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18237 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.002641395 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Pituitary.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 19047 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.002028925 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Rumen.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 16662 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.004742235 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Spleen.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18101 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9997 \n",
      "  * Corrected p-value threshold =  0.000496147 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Testis.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 21324 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.000427777 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Uterus.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18525 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.007263275 \n",
      "Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for tissue_file in ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/*.permutations.2rd.txt.gz; do\n",
    "    tissue_name=$(basename $tissue_file .permutations.2rd.txt.gz)\n",
    "    \n",
    "    Rscript ../../datasets/cGTEx/scripts/4_cis-eQTL_p-value.nominal.correction_basePermutation.r ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/${tissue_name}.permutations.2rd.txt.gz 0.05 ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/${tissue_name}.nominals.2rd.txt.gz ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/${tissue_name}.nominals.sig.txt\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "780d9738-d866-42fe-b671-22a7ddec5f15",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Adipose.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18057 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00279622 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Blood.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 16179 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9999 \n",
      "  * Corrected p-value threshold =  0.0221577 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Embryo.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 14097 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00559607 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Hypothalamus.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18588 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.001695535 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Intramuscular_fat.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 14928 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00741407 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Jejunum.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 17995 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00159072 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Kidney.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18967 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9996 \n",
      "  * Corrected p-value threshold =  0.000207602 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Leukocyte.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 14986 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.0006661515 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Liver.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 16558 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.0132995 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Lung.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 19125 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00488796 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Lymph_node.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18160 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00540915 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Macrophage.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 13753 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.0211331 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Mammary.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18222 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.006192955 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Mammary_L.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 16332 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.0262197 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Milk_cell.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 14881 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00121119 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Monocytes.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 15902 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.006728795 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Muscle.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 15451 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.0135557 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Muscle_cross.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 15497 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00258496 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Muscle_indicus.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 15291 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.000286652 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Muscle_taurus.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 15222 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00908362 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Ovary.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 17526 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.00177968 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Oviduct.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18237 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.002641395 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Pituitary.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 19047 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.002028925 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Rumen.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 16662 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.004742235 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Spleen.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18101 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9997 \n",
      "  * Corrected p-value threshold =  0.000496147 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Testis.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 21324 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.000427777 \n",
      "Done\n",
      "Processing fastQTL concatenated output [ ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/Uterus.permutations.2rd.txt.gz ] controlling for FDR = 5 %\n",
      "  * Number of molecular phenotypes = 18525 \n",
      "  * Correlation between Beta approx. and Empirical p-values = 0.9998 \n",
      "  * Corrected p-value threshold =  0.007263275 \n",
      "Done\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n",
      "Warning message:\n",
      "package ‘qvalue’ was built under R version 3.5.2 \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for tissue_file in ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/*.permutations.2rd.txt.gz; do\n",
    "    tissue_name=$(basename $tissue_file .permutations.2rd.txt.gz)\n",
    "    \n",
    "    Rscript ../../datasets/cGTEx/scripts/4_cis-eQTL_p-value.nominal.correction_basePermutation.MultipleTissueCorrection.r ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/${tissue_name}.permutations.2rd.txt.gz 0.05 ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/${tissue_name}.nominals.2rd.txt.gz ../../datasets/cGTEx/cis-eQTL_Summary_Statistics/significant_after_correction/${tissue_name}.nominals.sig.txt\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8018f422-0c91-4fa0-a73a-b9402ac1488c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ensembl Gene ID</th>\n",
       "      <th>Chromosome/scaffold name</th>\n",
       "      <th>Gene start (bp)</th>\n",
       "      <th>Gene end (bp)</th>\n",
       "      <th>Strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSBTAG00000006648</td>\n",
       "      <td>1</td>\n",
       "      <td>339070</td>\n",
       "      <td>350389</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSBTAG00000054829</td>\n",
       "      <td>1</td>\n",
       "      <td>626079</td>\n",
       "      <td>668340</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSBTAG00000001753</td>\n",
       "      <td>1</td>\n",
       "      <td>764121</td>\n",
       "      <td>820845</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSBTAG00000046015</td>\n",
       "      <td>1</td>\n",
       "      <td>141872518</td>\n",
       "      <td>141872892</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSBTAG00000021251</td>\n",
       "      <td>1</td>\n",
       "      <td>142011078</td>\n",
       "      <td>142037290</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSBTAG00000020035</td>\n",
       "      <td>1</td>\n",
       "      <td>882081</td>\n",
       "      <td>1002223</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENSBTAG00000005540</td>\n",
       "      <td>1</td>\n",
       "      <td>904168</td>\n",
       "      <td>904713</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENSBTAG00000001150</td>\n",
       "      <td>1</td>\n",
       "      <td>1040523</td>\n",
       "      <td>1047017</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENSBTAG00000026259</td>\n",
       "      <td>1</td>\n",
       "      <td>1061262</td>\n",
       "      <td>1061597</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSBTAG00000051226</td>\n",
       "      <td>1</td>\n",
       "      <td>1071552</td>\n",
       "      <td>1077244</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ENSBTAG00000013841</td>\n",
       "      <td>1</td>\n",
       "      <td>1099127</td>\n",
       "      <td>1100145</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENSBTAG00000021253</td>\n",
       "      <td>1</td>\n",
       "      <td>142065605</td>\n",
       "      <td>142132485</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSBTAG00000011528</td>\n",
       "      <td>1</td>\n",
       "      <td>1102887</td>\n",
       "      <td>1118312</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENSBTAG00000026260</td>\n",
       "      <td>1</td>\n",
       "      <td>1123943</td>\n",
       "      <td>1131748</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENSBTAG00000012594</td>\n",
       "      <td>1</td>\n",
       "      <td>1390803</td>\n",
       "      <td>1455219</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ENSBTAG00000021259</td>\n",
       "      <td>1</td>\n",
       "      <td>142142610</td>\n",
       "      <td>142204837</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENSBTAG00000018278</td>\n",
       "      <td>1</td>\n",
       "      <td>1643628</td>\n",
       "      <td>1651039</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ENSBTAG00000030836</td>\n",
       "      <td>1</td>\n",
       "      <td>142235364</td>\n",
       "      <td>142253631</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ENSBTAG00000021997</td>\n",
       "      <td>1</td>\n",
       "      <td>1665339</td>\n",
       "      <td>1908788</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ENSBTAG00000021270</td>\n",
       "      <td>1</td>\n",
       "      <td>142327757</td>\n",
       "      <td>142403210</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ENSBTAG00000008490</td>\n",
       "      <td>1</td>\n",
       "      <td>1909498</td>\n",
       "      <td>1943914</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ENSBTAG00000021272</td>\n",
       "      <td>1</td>\n",
       "      <td>142488705</td>\n",
       "      <td>142561127</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ENSBTAG00000035175</td>\n",
       "      <td>1</td>\n",
       "      <td>1942815</td>\n",
       "      <td>1954023</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ENSBTAG00000021276</td>\n",
       "      <td>1</td>\n",
       "      <td>142565466</td>\n",
       "      <td>142569246</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ENSBTAG00000030814</td>\n",
       "      <td>1</td>\n",
       "      <td>142614729</td>\n",
       "      <td>142618268</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ENSBTAG00000008482</td>\n",
       "      <td>1</td>\n",
       "      <td>1954230</td>\n",
       "      <td>1985068</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ENSBTAG00000048276</td>\n",
       "      <td>1</td>\n",
       "      <td>142629559</td>\n",
       "      <td>142634531</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ENSBTAG00000021280</td>\n",
       "      <td>1</td>\n",
       "      <td>142660182</td>\n",
       "      <td>142680495</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ENSBTAG00000023398</td>\n",
       "      <td>1</td>\n",
       "      <td>142694786</td>\n",
       "      <td>142737902</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ENSBTAG00000009188</td>\n",
       "      <td>1</td>\n",
       "      <td>1985522</td>\n",
       "      <td>2012383</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21600</th>\n",
       "      <td>ENSBTAG00000017761</td>\n",
       "      <td>X</td>\n",
       "      <td>90821253</td>\n",
       "      <td>90855974</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21601</th>\n",
       "      <td>ENSBTAG00000017769</td>\n",
       "      <td>X</td>\n",
       "      <td>90855707</td>\n",
       "      <td>90870094</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21602</th>\n",
       "      <td>ENSBTAG00000017779</td>\n",
       "      <td>X</td>\n",
       "      <td>90870392</td>\n",
       "      <td>90872909</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21603</th>\n",
       "      <td>ENSBTAG00000006122</td>\n",
       "      <td>X</td>\n",
       "      <td>90965898</td>\n",
       "      <td>91123238</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21604</th>\n",
       "      <td>ENSBTAG00000005098</td>\n",
       "      <td>X</td>\n",
       "      <td>91144490</td>\n",
       "      <td>91145083</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21605</th>\n",
       "      <td>ENSBTAG00000013289</td>\n",
       "      <td>X</td>\n",
       "      <td>91405559</td>\n",
       "      <td>91501300</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21606</th>\n",
       "      <td>ENSBTAG00000003855</td>\n",
       "      <td>X</td>\n",
       "      <td>91540004</td>\n",
       "      <td>91658010</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21607</th>\n",
       "      <td>ENSBTAG00000023182</td>\n",
       "      <td>X</td>\n",
       "      <td>91676599</td>\n",
       "      <td>91858715</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21608</th>\n",
       "      <td>ENSBTAG00000015858</td>\n",
       "      <td>X</td>\n",
       "      <td>91960664</td>\n",
       "      <td>91965617</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21609</th>\n",
       "      <td>ENSBTAG00000015860</td>\n",
       "      <td>X</td>\n",
       "      <td>91965590</td>\n",
       "      <td>92003575</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21610</th>\n",
       "      <td>ENSBTAG00000019370</td>\n",
       "      <td>X</td>\n",
       "      <td>92026276</td>\n",
       "      <td>92026920</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21611</th>\n",
       "      <td>ENSBTAG00000031564</td>\n",
       "      <td>X</td>\n",
       "      <td>92033596</td>\n",
       "      <td>92062014</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21612</th>\n",
       "      <td>ENSBTAG00000049441</td>\n",
       "      <td>X</td>\n",
       "      <td>92141442</td>\n",
       "      <td>92141597</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21613</th>\n",
       "      <td>ENSBTAG00000039888</td>\n",
       "      <td>X</td>\n",
       "      <td>92287353</td>\n",
       "      <td>92293899</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21614</th>\n",
       "      <td>ENSBTAG00000051604</td>\n",
       "      <td>X</td>\n",
       "      <td>92358756</td>\n",
       "      <td>92369249</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21615</th>\n",
       "      <td>ENSBTAG00000000172</td>\n",
       "      <td>X</td>\n",
       "      <td>92370954</td>\n",
       "      <td>92471753</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21616</th>\n",
       "      <td>ENSBTAG00000013177</td>\n",
       "      <td>X</td>\n",
       "      <td>92476249</td>\n",
       "      <td>92484051</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21617</th>\n",
       "      <td>ENSBTAG00000013178</td>\n",
       "      <td>X</td>\n",
       "      <td>92485200</td>\n",
       "      <td>92512358</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21618</th>\n",
       "      <td>ENSBTAG00000039090</td>\n",
       "      <td>X</td>\n",
       "      <td>92752215</td>\n",
       "      <td>92753650</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21619</th>\n",
       "      <td>ENSBTAG00000046659</td>\n",
       "      <td>X</td>\n",
       "      <td>92935902</td>\n",
       "      <td>92937829</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21620</th>\n",
       "      <td>ENSBTAG00000012980</td>\n",
       "      <td>X</td>\n",
       "      <td>93035620</td>\n",
       "      <td>93077042</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21621</th>\n",
       "      <td>ENSBTAG00000005852</td>\n",
       "      <td>X</td>\n",
       "      <td>93319321</td>\n",
       "      <td>93445155</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21622</th>\n",
       "      <td>ENSBTAG00000021843</td>\n",
       "      <td>X</td>\n",
       "      <td>93549709</td>\n",
       "      <td>93551583</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21623</th>\n",
       "      <td>ENSBTAG00000054746</td>\n",
       "      <td>X</td>\n",
       "      <td>93779678</td>\n",
       "      <td>93779884</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21624</th>\n",
       "      <td>ENSBTAG00000049019</td>\n",
       "      <td>X</td>\n",
       "      <td>93885325</td>\n",
       "      <td>93888238</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21625</th>\n",
       "      <td>ENSBTAG00000053877</td>\n",
       "      <td>X</td>\n",
       "      <td>93946743</td>\n",
       "      <td>93947359</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21626</th>\n",
       "      <td>ENSBTAG00000013260</td>\n",
       "      <td>X</td>\n",
       "      <td>94030462</td>\n",
       "      <td>94032343</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21627</th>\n",
       "      <td>ENSBTAG00000046769</td>\n",
       "      <td>X</td>\n",
       "      <td>94099596</td>\n",
       "      <td>94101552</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21628</th>\n",
       "      <td>ENSBTAG00000038387</td>\n",
       "      <td>X</td>\n",
       "      <td>94114121</td>\n",
       "      <td>94121116</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21629</th>\n",
       "      <td>ENSBTAG00000045616</td>\n",
       "      <td>X</td>\n",
       "      <td>94121459</td>\n",
       "      <td>94122571</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21630 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Ensembl Gene ID Chromosome/scaffold name  Gene start (bp)  \\\n",
       "0      ENSBTAG00000006648                        1           339070   \n",
       "1      ENSBTAG00000054829                        1           626079   \n",
       "2      ENSBTAG00000001753                        1           764121   \n",
       "3      ENSBTAG00000046015                        1        141872518   \n",
       "4      ENSBTAG00000021251                        1        142011078   \n",
       "5      ENSBTAG00000020035                        1           882081   \n",
       "6      ENSBTAG00000005540                        1           904168   \n",
       "7      ENSBTAG00000001150                        1          1040523   \n",
       "8      ENSBTAG00000026259                        1          1061262   \n",
       "9      ENSBTAG00000051226                        1          1071552   \n",
       "10     ENSBTAG00000013841                        1          1099127   \n",
       "11     ENSBTAG00000021253                        1        142065605   \n",
       "12     ENSBTAG00000011528                        1          1102887   \n",
       "13     ENSBTAG00000026260                        1          1123943   \n",
       "14     ENSBTAG00000012594                        1          1390803   \n",
       "15     ENSBTAG00000021259                        1        142142610   \n",
       "16     ENSBTAG00000018278                        1          1643628   \n",
       "17     ENSBTAG00000030836                        1        142235364   \n",
       "18     ENSBTAG00000021997                        1          1665339   \n",
       "19     ENSBTAG00000021270                        1        142327757   \n",
       "20     ENSBTAG00000008490                        1          1909498   \n",
       "21     ENSBTAG00000021272                        1        142488705   \n",
       "22     ENSBTAG00000035175                        1          1942815   \n",
       "23     ENSBTAG00000021276                        1        142565466   \n",
       "24     ENSBTAG00000030814                        1        142614729   \n",
       "25     ENSBTAG00000008482                        1          1954230   \n",
       "26     ENSBTAG00000048276                        1        142629559   \n",
       "27     ENSBTAG00000021280                        1        142660182   \n",
       "28     ENSBTAG00000023398                        1        142694786   \n",
       "29     ENSBTAG00000009188                        1          1985522   \n",
       "...                   ...                      ...              ...   \n",
       "21600  ENSBTAG00000017761                        X         90821253   \n",
       "21601  ENSBTAG00000017769                        X         90855707   \n",
       "21602  ENSBTAG00000017779                        X         90870392   \n",
       "21603  ENSBTAG00000006122                        X         90965898   \n",
       "21604  ENSBTAG00000005098                        X         91144490   \n",
       "21605  ENSBTAG00000013289                        X         91405559   \n",
       "21606  ENSBTAG00000003855                        X         91540004   \n",
       "21607  ENSBTAG00000023182                        X         91676599   \n",
       "21608  ENSBTAG00000015858                        X         91960664   \n",
       "21609  ENSBTAG00000015860                        X         91965590   \n",
       "21610  ENSBTAG00000019370                        X         92026276   \n",
       "21611  ENSBTAG00000031564                        X         92033596   \n",
       "21612  ENSBTAG00000049441                        X         92141442   \n",
       "21613  ENSBTAG00000039888                        X         92287353   \n",
       "21614  ENSBTAG00000051604                        X         92358756   \n",
       "21615  ENSBTAG00000000172                        X         92370954   \n",
       "21616  ENSBTAG00000013177                        X         92476249   \n",
       "21617  ENSBTAG00000013178                        X         92485200   \n",
       "21618  ENSBTAG00000039090                        X         92752215   \n",
       "21619  ENSBTAG00000046659                        X         92935902   \n",
       "21620  ENSBTAG00000012980                        X         93035620   \n",
       "21621  ENSBTAG00000005852                        X         93319321   \n",
       "21622  ENSBTAG00000021843                        X         93549709   \n",
       "21623  ENSBTAG00000054746                        X         93779678   \n",
       "21624  ENSBTAG00000049019                        X         93885325   \n",
       "21625  ENSBTAG00000053877                        X         93946743   \n",
       "21626  ENSBTAG00000013260                        X         94030462   \n",
       "21627  ENSBTAG00000046769                        X         94099596   \n",
       "21628  ENSBTAG00000038387                        X         94114121   \n",
       "21629  ENSBTAG00000045616                        X         94121459   \n",
       "\n",
       "       Gene end (bp)  Strand  \n",
       "0             350389      -1  \n",
       "1             668340       1  \n",
       "2             820845      -1  \n",
       "3          141872892       1  \n",
       "4          142037290      -1  \n",
       "5            1002223       1  \n",
       "6             904713      -1  \n",
       "7            1047017       1  \n",
       "8            1061597       1  \n",
       "9            1077244       1  \n",
       "10           1100145      -1  \n",
       "11         142132485      -1  \n",
       "12           1118312      -1  \n",
       "13           1131748      -1  \n",
       "14           1455219      -1  \n",
       "15         142204837      -1  \n",
       "16           1651039       1  \n",
       "17         142253631      -1  \n",
       "18           1908788      -1  \n",
       "19         142403210       1  \n",
       "20           1943914       1  \n",
       "21         142561127       1  \n",
       "22           1954023       1  \n",
       "23         142569246      -1  \n",
       "24         142618268      -1  \n",
       "25           1985068      -1  \n",
       "26         142634531      -1  \n",
       "27         142680495      -1  \n",
       "28         142737902       1  \n",
       "29           2012383       1  \n",
       "...              ...     ...  \n",
       "21600       90855974      -1  \n",
       "21601       90870094       1  \n",
       "21602       90872909      -1  \n",
       "21603       91123238      -1  \n",
       "21604       91145083       1  \n",
       "21605       91501300      -1  \n",
       "21606       91658010      -1  \n",
       "21607       91858715      -1  \n",
       "21608       91965617       1  \n",
       "21609       92003575      -1  \n",
       "21610       92026920       1  \n",
       "21611       92062014       1  \n",
       "21612       92141597      -1  \n",
       "21613       92293899       1  \n",
       "21614       92369249       1  \n",
       "21615       92471753      -1  \n",
       "21616       92484051       1  \n",
       "21617       92512358      -1  \n",
       "21618       92753650       1  \n",
       "21619       92937829       1  \n",
       "21620       93077042       1  \n",
       "21621       93445155       1  \n",
       "21622       93551583       1  \n",
       "21623       93779884       1  \n",
       "21624       93888238      -1  \n",
       "21625       93947359      -1  \n",
       "21626       94032343      -1  \n",
       "21627       94101552      -1  \n",
       "21628       94121116      -1  \n",
       "21629       94122571       1  \n",
       "\n",
       "[21630 rows x 5 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PCGenes = pd.read_csv('../../datasets/geneLists/Ensembl/CowV96/EnsV96CowProteinCodingGenes1-X.txt', sep='\\t')\n",
    "PCGenes.rename(columns={'Gene stable ID': 'Ensembl Gene ID'}, inplace=True)\n",
    "PCGenes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8037c4bd-9314-418b-878d-dc3900416b5e",
   "metadata": {},
   "outputs": [],
   "source": [
    "result = pd.DataFrame()\n",
    "for tissue in tissues:\n",
    "    tissueFrame = pd.read_csv('../../datasets/cGTEx/cis-eQTL_Summary_Statistics/'+tissue+'.permutations.2rd.txt.gz',\n",
    "            sep='\\s+',\n",
    "            names=['Ensembl Gene ID',\n",
    "                    'num_variants',\n",
    "                    'MLE_shape1',\n",
    "                    'MLE_shape2',\n",
    "                    'dummy',\n",
    "                    'best_variant',\n",
    "                    'distance',\n",
    "                    'nominal_pval',\n",
    "                    'slope',\n",
    "                    'corrected_pval_1',\n",
    "                    'corrected_pval_2'])\n",
    "    result = pd.concat([result, tissueFrame])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b8f212fa-cd3f-4d54-a877-69ef1149bbb9",
   "metadata": {},
   "outputs": [],
   "source": [
    "geneTissueExpressionCounts = result['Ensembl Gene ID'].value_counts().to_frame()\n",
    "geneTissueExpressionCounts.rename(columns={'Ensembl Gene ID':'expressedTissues'}, inplace=True)\n",
    "geneTissueExpressionCounts['Ensembl Gene ID'] = geneTissueExpressionCounts.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "9d3477f1-363c-43f3-a7b5-be28465ef824",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>expressedTissues</th>\n",
       "      <th>Ensembl Gene ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000008414</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000008414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000006129</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000006129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000006086</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000006086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000055051</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000055051</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000032433</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000032433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000001160</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000001160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000016139</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000016139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000002835</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000002835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000018970</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000018970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000051235</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000051235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000003168</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000003168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000019177</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000019177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000010254</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000010254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000009251</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000009251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000046837</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000046837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000034662</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000034662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000049036</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000049036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000013761</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000013761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000008034</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000008034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000008065</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000008065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000040295</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000040295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000013025</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000013025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000032223</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000032223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000005166</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000005166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000012371</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000012371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000030198</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000030198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000015481</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000015481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000012148</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000012148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000001123</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000001123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000005766</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000005766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000049304</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000049304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000004105</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000004105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000054627</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000054627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000042790</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000042790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000045085</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000045085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000053724</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000053724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000049458</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000049458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000053173</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000053173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000050629</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000050629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000039797</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000039797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000006304</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000006304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000050589</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000050589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000029761</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000029761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000045602</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000045602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000048465</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000048465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000015550</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000015550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000052506</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000052506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000047315</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000047315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000051945</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000051945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000043370</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000043370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000049899</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000049899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000054154</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000054154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000044892</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000044892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000055169</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000055169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000043625</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000043625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000048902</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000048902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000049687</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000049687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000008641</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000008641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000019732</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000019732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000050507</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000050507</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23513 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    expressedTissues     Ensembl Gene ID\n",
       "ENSBTAG00000008414                23  ENSBTAG00000008414\n",
       "ENSBTAG00000006129                23  ENSBTAG00000006129\n",
       "ENSBTAG00000006086                23  ENSBTAG00000006086\n",
       "ENSBTAG00000055051                23  ENSBTAG00000055051\n",
       "ENSBTAG00000032433                23  ENSBTAG00000032433\n",
       "ENSBTAG00000001160                23  ENSBTAG00000001160\n",
       "ENSBTAG00000016139                23  ENSBTAG00000016139\n",
       "ENSBTAG00000002835                23  ENSBTAG00000002835\n",
       "ENSBTAG00000018970                23  ENSBTAG00000018970\n",
       "ENSBTAG00000051235                23  ENSBTAG00000051235\n",
       "ENSBTAG00000003168                23  ENSBTAG00000003168\n",
       "ENSBTAG00000019177                23  ENSBTAG00000019177\n",
       "ENSBTAG00000010254                23  ENSBTAG00000010254\n",
       "ENSBTAG00000009251                23  ENSBTAG00000009251\n",
       "ENSBTAG00000046837                23  ENSBTAG00000046837\n",
       "ENSBTAG00000034662                23  ENSBTAG00000034662\n",
       "ENSBTAG00000049036                23  ENSBTAG00000049036\n",
       "ENSBTAG00000013761                23  ENSBTAG00000013761\n",
       "ENSBTAG00000008034                23  ENSBTAG00000008034\n",
       "ENSBTAG00000008065                23  ENSBTAG00000008065\n",
       "ENSBTAG00000040295                23  ENSBTAG00000040295\n",
       "ENSBTAG00000013025                23  ENSBTAG00000013025\n",
       "ENSBTAG00000032223                23  ENSBTAG00000032223\n",
       "ENSBTAG00000005166                23  ENSBTAG00000005166\n",
       "ENSBTAG00000012371                23  ENSBTAG00000012371\n",
       "ENSBTAG00000030198                23  ENSBTAG00000030198\n",
       "ENSBTAG00000015481                23  ENSBTAG00000015481\n",
       "ENSBTAG00000012148                23  ENSBTAG00000012148\n",
       "ENSBTAG00000001123                23  ENSBTAG00000001123\n",
       "ENSBTAG00000005766                23  ENSBTAG00000005766\n",
       "...                              ...                 ...\n",
       "ENSBTAG00000049304                 1  ENSBTAG00000049304\n",
       "ENSBTAG00000004105                 1  ENSBTAG00000004105\n",
       "ENSBTAG00000054627                 1  ENSBTAG00000054627\n",
       "ENSBTAG00000042790                 1  ENSBTAG00000042790\n",
       "ENSBTAG00000045085                 1  ENSBTAG00000045085\n",
       "ENSBTAG00000053724                 1  ENSBTAG00000053724\n",
       "ENSBTAG00000049458                 1  ENSBTAG00000049458\n",
       "ENSBTAG00000053173                 1  ENSBTAG00000053173\n",
       "ENSBTAG00000050629                 1  ENSBTAG00000050629\n",
       "ENSBTAG00000039797                 1  ENSBTAG00000039797\n",
       "ENSBTAG00000006304                 1  ENSBTAG00000006304\n",
       "ENSBTAG00000050589                 1  ENSBTAG00000050589\n",
       "ENSBTAG00000029761                 1  ENSBTAG00000029761\n",
       "ENSBTAG00000045602                 1  ENSBTAG00000045602\n",
       "ENSBTAG00000048465                 1  ENSBTAG00000048465\n",
       "ENSBTAG00000015550                 1  ENSBTAG00000015550\n",
       "ENSBTAG00000052506                 1  ENSBTAG00000052506\n",
       "ENSBTAG00000047315                 1  ENSBTAG00000047315\n",
       "ENSBTAG00000051945                 1  ENSBTAG00000051945\n",
       "ENSBTAG00000043370                 1  ENSBTAG00000043370\n",
       "ENSBTAG00000049899                 1  ENSBTAG00000049899\n",
       "ENSBTAG00000054154                 1  ENSBTAG00000054154\n",
       "ENSBTAG00000044892                 1  ENSBTAG00000044892\n",
       "ENSBTAG00000055169                 1  ENSBTAG00000055169\n",
       "ENSBTAG00000043625                 1  ENSBTAG00000043625\n",
       "ENSBTAG00000048902                 1  ENSBTAG00000048902\n",
       "ENSBTAG00000049687                 1  ENSBTAG00000049687\n",
       "ENSBTAG00000008641                 1  ENSBTAG00000008641\n",
       "ENSBTAG00000019732                 1  ENSBTAG00000019732\n",
       "ENSBTAG00000050507                 1  ENSBTAG00000050507\n",
       "\n",
       "[23513 rows x 2 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "geneTissueExpressionCounts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3fc6c312-00ca-4f3f-b8b2-59c818da0110",
   "metadata": {},
   "outputs": [],
   "source": [
    "result = pd.DataFrame()\n",
    "for tissue in tissues:\n",
    "    tissueFrame = pd.read_csv('../../datasets/cGTEx/cis-eQTL_Summary_Statistics/significant_after_correction/BonferroniCorrectedToo/'+tissue+'.nominals.sig.txt', sep='\\t')\n",
    "    tissueFrame = pd.DataFrame(tissueFrame['pid'].unique())\n",
    "    result = pd.concat([result, tissueFrame])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "15e06e89-c4c3-4b03-a13b-f6dc2142e296",
   "metadata": {},
   "outputs": [],
   "source": [
    "geneTissueeQTLCounts = result[0].value_counts().to_frame()\n",
    "geneTissueeQTLCounts.rename(columns={0:'affectedTissues'}, inplace=True)\n",
    "geneTissueeQTLCounts['Ensembl Gene ID'] = geneTissueeQTLCounts.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f31b3583-1cf0-43e9-9e64-38a4d3f57e8a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>affectedTissues</th>\n",
       "      <th>Ensembl Gene ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000037605</th>\n",
       "      <td>20</td>\n",
       "      <td>ENSBTAG00000037605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000021077</th>\n",
       "      <td>19</td>\n",
       "      <td>ENSBTAG00000021077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000007075</th>\n",
       "      <td>18</td>\n",
       "      <td>ENSBTAG00000007075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000039664</th>\n",
       "      <td>17</td>\n",
       "      <td>ENSBTAG00000039664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000053807</th>\n",
       "      <td>17</td>\n",
       "      <td>ENSBTAG00000053807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000052397</th>\n",
       "      <td>17</td>\n",
       "      <td>ENSBTAG00000052397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000002477</th>\n",
       "      <td>17</td>\n",
       "      <td>ENSBTAG00000002477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000047314</th>\n",
       "      <td>17</td>\n",
       "      <td>ENSBTAG00000047314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000050846</th>\n",
       "      <td>17</td>\n",
       "      <td>ENSBTAG00000050846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000054241</th>\n",
       "      <td>17</td>\n",
       "      <td>ENSBTAG00000054241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000021565</th>\n",
       "      <td>17</td>\n",
       "      <td>ENSBTAG00000021565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000009656</th>\n",
       "      <td>17</td>\n",
       "      <td>ENSBTAG00000009656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000005868</th>\n",
       "      <td>17</td>\n",
       "      <td>ENSBTAG00000005868</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000053632</th>\n",
       "      <td>16</td>\n",
       "      <td>ENSBTAG00000053632</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000005715</th>\n",
       "      <td>16</td>\n",
       "      <td>ENSBTAG00000005715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000038112</th>\n",
       "      <td>16</td>\n",
       "      <td>ENSBTAG00000038112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000019588</th>\n",
       "      <td>16</td>\n",
       "      <td>ENSBTAG00000019588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000049294</th>\n",
       "      <td>16</td>\n",
       "      <td>ENSBTAG00000049294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000015727</th>\n",
       "      <td>16</td>\n",
       "      <td>ENSBTAG00000015727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000011019</th>\n",
       "      <td>15</td>\n",
       "      <td>ENSBTAG00000011019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000037937</th>\n",
       "      <td>15</td>\n",
       "      <td>ENSBTAG00000037937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000015166</th>\n",
       "      <td>15</td>\n",
       "      <td>ENSBTAG00000015166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000045929</th>\n",
       "      <td>15</td>\n",
       "      <td>ENSBTAG00000045929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000015853</th>\n",
       "      <td>15</td>\n",
       "      <td>ENSBTAG00000015853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000024272</th>\n",
       "      <td>15</td>\n",
       "      <td>ENSBTAG00000024272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000004815</th>\n",
       "      <td>15</td>\n",
       "      <td>ENSBTAG00000004815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000014768</th>\n",
       "      <td>15</td>\n",
       "      <td>ENSBTAG00000014768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000002786</th>\n",
       "      <td>15</td>\n",
       "      <td>ENSBTAG00000002786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000019569</th>\n",
       "      <td>15</td>\n",
       "      <td>ENSBTAG00000019569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000027899</th>\n",
       "      <td>15</td>\n",
       "      <td>ENSBTAG00000027899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000044118</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000044118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000000679</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000000679</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000026851</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000026851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000017490</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000017490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000055222</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000055222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000053735</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000053735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000003721</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000003721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000007721</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000007721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000015242</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000015242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000054317</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000054317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000001080</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000001080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000018456</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000018456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000006014</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000006014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000015874</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000015874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000052466</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000052466</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000004448</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000004448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000010897</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000010897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000009291</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000009291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000022036</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000022036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000003232</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000003232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000000808</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000000808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000021875</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000021875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000018868</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000018868</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000006111</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000006111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000020272</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000020272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000004161</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000004161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000001920</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000001920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000005644</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000005644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000053442</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000053442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSBTAG00000048764</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000048764</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13143 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    affectedTissues     Ensembl Gene ID\n",
       "ENSBTAG00000037605               20  ENSBTAG00000037605\n",
       "ENSBTAG00000021077               19  ENSBTAG00000021077\n",
       "ENSBTAG00000007075               18  ENSBTAG00000007075\n",
       "ENSBTAG00000039664               17  ENSBTAG00000039664\n",
       "ENSBTAG00000053807               17  ENSBTAG00000053807\n",
       "ENSBTAG00000052397               17  ENSBTAG00000052397\n",
       "ENSBTAG00000002477               17  ENSBTAG00000002477\n",
       "ENSBTAG00000047314               17  ENSBTAG00000047314\n",
       "ENSBTAG00000050846               17  ENSBTAG00000050846\n",
       "ENSBTAG00000054241               17  ENSBTAG00000054241\n",
       "ENSBTAG00000021565               17  ENSBTAG00000021565\n",
       "ENSBTAG00000009656               17  ENSBTAG00000009656\n",
       "ENSBTAG00000005868               17  ENSBTAG00000005868\n",
       "ENSBTAG00000053632               16  ENSBTAG00000053632\n",
       "ENSBTAG00000005715               16  ENSBTAG00000005715\n",
       "ENSBTAG00000038112               16  ENSBTAG00000038112\n",
       "ENSBTAG00000019588               16  ENSBTAG00000019588\n",
       "ENSBTAG00000049294               16  ENSBTAG00000049294\n",
       "ENSBTAG00000015727               16  ENSBTAG00000015727\n",
       "ENSBTAG00000011019               15  ENSBTAG00000011019\n",
       "ENSBTAG00000037937               15  ENSBTAG00000037937\n",
       "ENSBTAG00000015166               15  ENSBTAG00000015166\n",
       "ENSBTAG00000045929               15  ENSBTAG00000045929\n",
       "ENSBTAG00000015853               15  ENSBTAG00000015853\n",
       "ENSBTAG00000024272               15  ENSBTAG00000024272\n",
       "ENSBTAG00000004815               15  ENSBTAG00000004815\n",
       "ENSBTAG00000014768               15  ENSBTAG00000014768\n",
       "ENSBTAG00000002786               15  ENSBTAG00000002786\n",
       "ENSBTAG00000019569               15  ENSBTAG00000019569\n",
       "ENSBTAG00000027899               15  ENSBTAG00000027899\n",
       "...                             ...                 ...\n",
       "ENSBTAG00000044118                1  ENSBTAG00000044118\n",
       "ENSBTAG00000000679                1  ENSBTAG00000000679\n",
       "ENSBTAG00000026851                1  ENSBTAG00000026851\n",
       "ENSBTAG00000017490                1  ENSBTAG00000017490\n",
       "ENSBTAG00000055222                1  ENSBTAG00000055222\n",
       "ENSBTAG00000053735                1  ENSBTAG00000053735\n",
       "ENSBTAG00000003721                1  ENSBTAG00000003721\n",
       "ENSBTAG00000007721                1  ENSBTAG00000007721\n",
       "ENSBTAG00000015242                1  ENSBTAG00000015242\n",
       "ENSBTAG00000054317                1  ENSBTAG00000054317\n",
       "ENSBTAG00000001080                1  ENSBTAG00000001080\n",
       "ENSBTAG00000018456                1  ENSBTAG00000018456\n",
       "ENSBTAG00000006014                1  ENSBTAG00000006014\n",
       "ENSBTAG00000015874                1  ENSBTAG00000015874\n",
       "ENSBTAG00000052466                1  ENSBTAG00000052466\n",
       "ENSBTAG00000004448                1  ENSBTAG00000004448\n",
       "ENSBTAG00000010897                1  ENSBTAG00000010897\n",
       "ENSBTAG00000009291                1  ENSBTAG00000009291\n",
       "ENSBTAG00000022036                1  ENSBTAG00000022036\n",
       "ENSBTAG00000003232                1  ENSBTAG00000003232\n",
       "ENSBTAG00000000808                1  ENSBTAG00000000808\n",
       "ENSBTAG00000021875                1  ENSBTAG00000021875\n",
       "ENSBTAG00000018868                1  ENSBTAG00000018868\n",
       "ENSBTAG00000006111                1  ENSBTAG00000006111\n",
       "ENSBTAG00000020272                1  ENSBTAG00000020272\n",
       "ENSBTAG00000004161                1  ENSBTAG00000004161\n",
       "ENSBTAG00000001920                1  ENSBTAG00000001920\n",
       "ENSBTAG00000005644                1  ENSBTAG00000005644\n",
       "ENSBTAG00000053442                1  ENSBTAG00000053442\n",
       "ENSBTAG00000048764                1  ENSBTAG00000048764\n",
       "\n",
       "[13143 rows x 2 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "geneTissueeQTLCounts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4331622f-18ee-438f-9cc6-5f83a72651ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>expressedTissues</th>\n",
       "      <th>Ensembl Gene ID</th>\n",
       "      <th>affectedTissues</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000008414</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000006129</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000006086</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000055051</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000032433</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000001160</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000016139</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000002835</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000018970</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000051235</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000003168</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000019177</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000010254</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000009251</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000046837</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000034662</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000049036</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000013761</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000008034</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000008065</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000040295</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000013025</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000032223</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000005166</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000012371</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000030198</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000015481</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000012148</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000001123</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>23</td>\n",
       "      <td>ENSBTAG00000005766</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23483</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000049304</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23484</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000004105</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23485</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000054627</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23486</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000042790</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23487</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000045085</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23488</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000053724</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23489</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000049458</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23490</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000053173</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23491</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000050629</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23492</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000039797</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23493</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000006304</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23494</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000050589</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23495</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000029761</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23496</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000045602</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23497</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000048465</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23498</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000015550</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23499</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000052506</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23500</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000047315</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23501</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000051945</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23502</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000043370</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23503</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000049899</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23504</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000054154</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23505</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000044892</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23506</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000055169</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23507</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000043625</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23508</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000048902</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23509</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000049687</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23510</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000008641</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23511</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000019732</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23512</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSBTAG00000050507</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23513 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       expressedTissues     Ensembl Gene ID  affectedTissues\n",
       "0                    23  ENSBTAG00000008414              2.0\n",
       "1                    23  ENSBTAG00000006129              2.0\n",
       "2                    23  ENSBTAG00000006086              3.0\n",
       "3                    23  ENSBTAG00000055051              2.0\n",
       "4                    23  ENSBTAG00000032433              1.0\n",
       "5                    23  ENSBTAG00000001160              0.0\n",
       "6                    23  ENSBTAG00000016139              2.0\n",
       "7                    23  ENSBTAG00000002835              3.0\n",
       "8                    23  ENSBTAG00000018970              0.0\n",
       "9                    23  ENSBTAG00000051235              0.0\n",
       "10                   23  ENSBTAG00000003168              1.0\n",
       "11                   23  ENSBTAG00000019177              1.0\n",
       "12                   23  ENSBTAG00000010254              1.0\n",
       "13                   23  ENSBTAG00000009251              2.0\n",
       "14                   23  ENSBTAG00000046837              5.0\n",
       "15                   23  ENSBTAG00000034662              4.0\n",
       "16                   23  ENSBTAG00000049036              2.0\n",
       "17                   23  ENSBTAG00000013761              1.0\n",
       "18                   23  ENSBTAG00000008034              1.0\n",
       "19                   23  ENSBTAG00000008065             11.0\n",
       "20                   23  ENSBTAG00000040295              2.0\n",
       "21                   23  ENSBTAG00000013025              1.0\n",
       "22                   23  ENSBTAG00000032223              2.0\n",
       "23                   23  ENSBTAG00000005166              1.0\n",
       "24                   23  ENSBTAG00000012371              0.0\n",
       "25                   23  ENSBTAG00000030198              4.0\n",
       "26                   23  ENSBTAG00000015481              2.0\n",
       "27                   23  ENSBTAG00000012148              1.0\n",
       "28                   23  ENSBTAG00000001123              0.0\n",
       "29                   23  ENSBTAG00000005766              4.0\n",
       "...                 ...                 ...              ...\n",
       "23483                 1  ENSBTAG00000049304              1.0\n",
       "23484                 1  ENSBTAG00000004105              0.0\n",
       "23485                 1  ENSBTAG00000054627              0.0\n",
       "23486                 1  ENSBTAG00000042790              0.0\n",
       "23487                 1  ENSBTAG00000045085              0.0\n",
       "23488                 1  ENSBTAG00000053724              0.0\n",
       "23489                 1  ENSBTAG00000049458              0.0\n",
       "23490                 1  ENSBTAG00000053173              0.0\n",
       "23491                 1  ENSBTAG00000050629              0.0\n",
       "23492                 1  ENSBTAG00000039797              0.0\n",
       "23493                 1  ENSBTAG00000006304              0.0\n",
       "23494                 1  ENSBTAG00000050589              0.0\n",
       "23495                 1  ENSBTAG00000029761              0.0\n",
       "23496                 1  ENSBTAG00000045602              0.0\n",
       "23497                 1  ENSBTAG00000048465              0.0\n",
       "23498                 1  ENSBTAG00000015550              0.0\n",
       "23499                 1  ENSBTAG00000052506              0.0\n",
       "23500                 1  ENSBTAG00000047315              0.0\n",
       "23501                 1  ENSBTAG00000051945              0.0\n",
       "23502                 1  ENSBTAG00000043370              0.0\n",
       "23503                 1  ENSBTAG00000049899              0.0\n",
       "23504                 1  ENSBTAG00000054154              0.0\n",
       "23505                 1  ENSBTAG00000044892              0.0\n",
       "23506                 1  ENSBTAG00000055169              0.0\n",
       "23507                 1  ENSBTAG00000043625              0.0\n",
       "23508                 1  ENSBTAG00000048902              0.0\n",
       "23509                 1  ENSBTAG00000049687              0.0\n",
       "23510                 1  ENSBTAG00000008641              0.0\n",
       "23511                 1  ENSBTAG00000019732              0.0\n",
       "23512                 1  ENSBTAG00000050507              0.0\n",
       "\n",
       "[23513 rows x 3 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "geneTissueExpressionCounts = pd.merge(geneTissueExpressionCounts, geneTissueeQTLCounts, on='Ensembl Gene ID', how='left').fillna(0)\n",
    "geneTissueExpressionCounts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f9cb5a32-9d36-4f21-8616-eabbe9d6b4d2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19423"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "geneTissueExpressionCounts = pd.merge(geneTissueExpressionCounts, PCGenes, on='Ensembl Gene ID')\n",
    "len(geneTissueExpressionCounts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "93cfc11b-efda-470c-95d0-13e38af0165f",
   "metadata": {},
   "outputs": [],
   "source": [
    "geneTissueExpressionCounts.to_csv('../../outputFiles/cGTExGenesWithCounts.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:paper-eqtls]",
   "language": "python",
   "name": "conda-env-paper-eqtls-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
