{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import GTEx data\n",
    "Add GTEx v7 data to MySQL database"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Housekeeping"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Background\n",
    "\n",
    "* GTEx tested for eQTLs in each tissue separately\n",
    "* Number of tissues tested for eQTLs needs to be adjusted for multiple testing\n",
    "* Count where Bonferroni adjusted p-values are significant and also eQTLs that meet Metasoft threshold (>0.9)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import mysql.connector\n",
    "from sqlalchemy import create_engine\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "engine = create_engine('mysql+mysqlconnector://jupyter:password@localhost:3306/gtex', echo=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove version numbers from Ensembl Gene IDs e.g. 'ENSG000000001.8' to 'ENSG000000001'. Enables comparison between IDs from different sources"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def removeGeneIDVersions(text):\n",
    "    return re.findall('(ENSG\\d+)', text)[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Constants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### List of GTEx tissues"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "tissues = ['Adipose - Subcutaneous',\n",
    " 'Adipose - Visceral (Omentum)',\n",
    " 'Adrenal Gland',\n",
    " 'Artery - Aorta',\n",
    " 'Artery - Coronary',\n",
    " 'Artery - Tibial',\n",
    " 'Brain - Amygdala',\n",
    " 'Brain - Anterior cingulate cortex (BA24)',\n",
    " 'Brain - Caudate (basal ganglia)',\n",
    " 'Brain - Cerebellar Hemisphere',\n",
    " 'Brain - Cerebellum',\n",
    " 'Brain - Cortex',\n",
    " 'Brain - Frontal Cortex (BA9)',\n",
    " 'Brain - Hippocampus',\n",
    " 'Brain - Hypothalamus',\n",
    " 'Brain - Nucleus accumbens (basal ganglia)',\n",
    " 'Brain - Putamen (basal ganglia)',\n",
    " 'Brain - Spinal cord (cervical c-1)',\n",
    " 'Brain - Substantia nigra',\n",
    " 'Breast - Mammary Tissue',\n",
    " 'Cells - EBV-transformed lymphocytes',\n",
    " 'Cells - Transformed fibroblasts',\n",
    " 'Colon - Sigmoid',\n",
    " 'Colon - Transverse',\n",
    " 'Esophagus - Gastroesophageal Junction',\n",
    " 'Esophagus - Mucosa',\n",
    " 'Esophagus - Muscularis',\n",
    " 'Heart - Atrial Appendage',\n",
    " 'Heart - Left Ventricle',\n",
    " 'Liver',\n",
    " 'Lung',\n",
    " 'Minor Salivary Gland',\n",
    " 'Muscle - Skeletal',\n",
    " 'Nerve - Tibial',\n",
    " 'Ovary',\n",
    " 'Pancreas',\n",
    " 'Pituitary',\n",
    " 'Prostate',\n",
    " 'Skin - Not Sun Exposed (Suprapubic)',\n",
    " 'Skin - Sun Exposed (Lower leg)',\n",
    " 'Small Intestine - Terminal Ileum',\n",
    " 'Spleen',\n",
    " 'Stomach',\n",
    " 'Testis',\n",
    " 'Thyroid',\n",
    " 'Uterus',\n",
    " 'Vagina',\n",
    " 'Whole Blood']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Syntax to create MySQL tables\n",
    "\n",
    "```sql\n",
    "CREATE TABLE `v7` (\n",
    "  `variant_id` varchar(255) DEFAULT NULL,\n",
    "  `gene_id` varchar(255) DEFAULT NULL,\n",
    "  `tss_distance` int(11) DEFAULT NULL,\n",
    "  `ma_samples` int(11) DEFAULT NULL,\n",
    "  `ma_count` int(11) DEFAULT NULL,\n",
    "  `maf` double DEFAULT NULL,\n",
    "  `pval_nominal` double DEFAULT NULL,\n",
    "  `slope` double DEFAULT NULL,\n",
    "  `slope_se` double DEFAULT NULL,\n",
    "  `pval_nominal_threshold` double DEFAULT NULL,\n",
    "  `min_pval_nominal` double DEFAULT NULL,\n",
    "  `pval_beta` double DEFAULT NULL,\n",
    "  `tissue` varchar(255) DEFAULT NULL,\n",
    "  `sigAfterBonferroni` tinyint(1) DEFAULT NULL,\n",
    "  KEY `tissue_gene_id` (`tissue`,`gene_id`),\n",
    "  KEY `sigAfterBonferroni` (`sigAfterBonferroni`),\n",
    "  KEY `sigAfterBonferroni_2` (`sigAfterBonferroni`,`gene_id`,`tissue`)\n",
    ") ENGINE=InnoDB DEFAULT CHARSET=utf8;\n",
    "```\n",
    "\n",
    "```sql\n",
    "CREATE TABLE `v7Metasoft` (\n",
    "  `RSID` varchar(300) NOT NULL DEFAULT '',\n",
    "  `snp` varchar(300) NOT NULL,\n",
    "  `gene` varchar(300) NOT NULL,\n",
    "  `#STUDY` tinyint(2) DEFAULT NULL,\n",
    "  `PVALUE_FE` double DEFAULT NULL,\n",
    "  `BETA_FE` decimal(20,15) DEFAULT NULL,\n",
    "  `STD_FE` decimal(20,15) DEFAULT NULL,\n",
    "  `PVALUE_RE` double DEFAULT NULL,\n",
    "  `BETA_RE` decimal(20,15) DEFAULT NULL,\n",
    "  `STD_RE` decimal(20,15) DEFAULT NULL,\n",
    "  `PVALUE_RE2` double DEFAULT NULL,\n",
    "  `STAT1_RE2` decimal(20,13) DEFAULT NULL,\n",
    "  `STAT2_RE2` decimal(20,13) DEFAULT NULL,\n",
    "  `PVALUE_BE` double DEFAULT NULL,\n",
    "  `I_SQUARE` decimal(20,13) DEFAULT NULL,\n",
    "  `Q` decimal(20,13) DEFAULT NULL,\n",
    "  `PVALUE_Q` double DEFAULT NULL,\n",
    "  `TAU_SQUARE` decimal(20,15) DEFAULT NULL,\n",
    "  `mvalTissues` tinyint(2) DEFAULT NULL,\n",
    "  `pval_Adipose_Subcutaneous` double DEFAULT NULL,\n",
    "  `pval_Adipose_Visceral_Omentum` double DEFAULT NULL,\n",
    "  `pval_Adrenal_Gland` double DEFAULT NULL,\n",
    "  `pval_Artery_Aorta` double DEFAULT NULL,\n",
    "  `pval_Artery_Coronary` double DEFAULT NULL,\n",
    "  `pval_Artery_Tibial` double DEFAULT NULL,\n",
    "  `pval_Brain_Amygdala` double DEFAULT NULL,\n",
    "  `pval_Brain_Anterior_cingulate_cortex_BA24` double DEFAULT NULL,\n",
    "  `pval_Brain_Caudate_basal_ganglia` double DEFAULT NULL,\n",
    "  `pval_Brain_Cerebellar_Hemisphere` double DEFAULT NULL,\n",
    "  `pval_Brain_Cerebellum` double DEFAULT NULL,\n",
    "  `pval_Brain_Cortex` double DEFAULT NULL,\n",
    "  `pval_Brain_Frontal_Cortex_BA9` double DEFAULT NULL,\n",
    "  `pval_Brain_Hippocampus` double DEFAULT NULL,\n",
    "  `pval_Brain_Hypothalamus` double DEFAULT NULL,\n",
    "  `pval_Brain_Nucleus_accumbens_basal_ganglia` double DEFAULT NULL,\n",
    "  `pval_Brain_Putamen_basal_ganglia` double DEFAULT NULL,\n",
    "  `pval_Brain_Spinal_cord_cervical_c-1` double DEFAULT NULL,\n",
    "  `pval_Brain_Substantia_nigra` double DEFAULT NULL,\n",
    "  `pval_Breast_Mammary_Tissue` double DEFAULT NULL,\n",
    "  `pval_Cells_EBV-transformed_lymphocytes` double DEFAULT NULL,\n",
    "  `pval_Cells_Transformed_fibroblasts` double DEFAULT NULL,\n",
    "  `pval_Colon_Sigmoid` double DEFAULT NULL,\n",
    "  `pval_Colon_Transverse` double DEFAULT NULL,\n",
    "  `pval_Esophagus_Gastroesophageal_Junction` double DEFAULT NULL,\n",
    "  `pval_Esophagus_Mucosa` double DEFAULT NULL,\n",
    "  `pval_Esophagus_Muscularis` double DEFAULT NULL,\n",
    "  `pval_Heart_Atrial_Appendage` double DEFAULT NULL,\n",
    "  `pval_Heart_Left_Ventricle` double DEFAULT NULL,\n",
    "  `pval_Liver` double DEFAULT NULL,\n",
    "  `pval_Lung` double DEFAULT NULL,\n",
    "  `pval_Minor_Salivary_Gland` double DEFAULT NULL,\n",
    "  `pval_Muscle_Skeletal` double DEFAULT NULL,\n",
    "  `pval_Nerve_Tibial` double DEFAULT NULL,\n",
    "  `pval_Ovary` double DEFAULT NULL,\n",
    "  `pval_Pancreas` double DEFAULT NULL,\n",
    "  `pval_Pituitary` double DEFAULT NULL,\n",
    "  `pval_Prostate` double DEFAULT NULL,\n",
    "  `pval_Skin_Not_Sun_Exposed_Suprapubic` double DEFAULT NULL,\n",
    "  `pval_Skin_Sun_Exposed_Lower_leg` double DEFAULT NULL,\n",
    "  `pval_Small_Intestine_Terminal_Ileum` double DEFAULT NULL,\n",
    "  `pval_Spleen` double DEFAULT NULL,\n",
    "  `pval_Stomach` double DEFAULT NULL,\n",
    "  `pval_Testis` double DEFAULT NULL,\n",
    "  `pval_Thyroid` double DEFAULT NULL,\n",
    "  `pval_Uterus` double DEFAULT NULL,\n",
    "  `pval_Vagina` double DEFAULT NULL,\n",
    "  `pval_Whole_Blood` double DEFAULT NULL,\n",
    "  `mval_Adipose_Subcutaneous` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Adipose_Visceral_Omentum` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Adrenal_Gland` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Artery_Aorta` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Artery_Coronary` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Artery_Tibial` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Brain_Amygdala` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Brain_Anterior_cingulate_cortex_BA24` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Brain_Caudate_basal_ganglia` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Brain_Cerebellar_Hemisphere` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Brain_Cerebellum` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Brain_Cortex` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Brain_Frontal_Cortex_BA9` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Brain_Hippocampus` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Brain_Hypothalamus` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Brain_Nucleus_accumbens_basal_ganglia` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Brain_Putamen_basal_ganglia` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Brain_Spinal_cord_cervical_c-1` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Brain_Substantia_nigra` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Breast_Mammary_Tissue` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Cells_EBV-transformed_lymphocytes` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Cells_Transformed_fibroblasts` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Colon_Sigmoid` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Colon_Transverse` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Esophagus_Gastroesophageal_Junction` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Esophagus_Mucosa` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Esophagus_Muscularis` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Heart_Atrial_Appendage` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Heart_Left_Ventricle` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Liver` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Lung` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Minor_Salivary_Gland` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Muscle_Skeletal` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Nerve_Tibial` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Ovary` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Pancreas` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Pituitary` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Prostate` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Skin_Not_Sun_Exposed_Suprapubic` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Skin_Sun_Exposed_Lower_leg` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Small_Intestine_Terminal_Ileum` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Spleen` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Stomach` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Testis` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Thyroid` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Uterus` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Vagina` decimal(4,3) DEFAULT NULL,\n",
    "  `mval_Whole_Blood` decimal(4,3) DEFAULT NULL\n",
    ") ENGINE=InnoDB DEFAULT CHARSET=utf8;\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Count number of tissues with Metasoft eQTL"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```sql\n",
    "UPDATE v7Metasoft\n",
    "SET mvalTissues =\n",
    "(IF(mval_Adipose_Subcutaneous > 0.9, 1, 0)\n",
    "+ IF(mval_Adipose_Visceral_Omentum > 0.9, 1, 0)\n",
    "+ IF(mval_Adrenal_Gland > 0.9, 1, 0)\n",
    "+ IF(mval_Artery_Aorta > 0.9, 1, 0)\n",
    "+ IF(mval_Artery_Coronary > 0.9, 1, 0)\n",
    "+ IF(mval_Artery_Tibial > 0.9, 1, 0)\n",
    "+ IF(mval_Brain_Amygdala > 0.9, 1, 0)\n",
    "+ IF(mval_Brain_Anterior_cingulate_cortex_BA24 > 0.9, 1, 0)\n",
    "+ IF(mval_Brain_Caudate_basal_ganglia > 0.9, 1, 0)\n",
    "+ IF(mval_Brain_Cerebellar_Hemisphere > 0.9, 1, 0)\n",
    "+ IF(mval_Brain_Cerebellum > 0.9, 1, 0)\n",
    "+ IF(mval_Brain_Cortex > 0.9, 1, 0)\n",
    "+ IF(mval_Brain_Frontal_Cortex_BA9 > 0.9, 1, 0)\n",
    "+ IF(mval_Brain_Hippocampus > 0.9, 1, 0)\n",
    "+ IF(mval_Brain_Hypothalamus > 0.9, 1, 0)\n",
    "+ IF(mval_Brain_Nucleus_accumbens_basal_ganglia > 0.9, 1, 0)\n",
    "+ IF(mval_Brain_Putamen_basal_ganglia > 0.9, 1, 0)\n",
    "+ IF(`mval_Brain_Spinal_cord_cervical_c-1` > 0.9, 1, 0)\n",
    "+ IF(mval_Brain_Substantia_nigra > 0.9, 1, 0)\n",
    "+ IF(mval_Breast_Mammary_Tissue > 0.9, 1, 0)\n",
    "+ IF(`mval_Cells_EBV-transformed_lymphocytes` > 0.9, 1, 0)\n",
    "+ IF(mval_Cells_Transformed_fibroblasts > 0.9, 1, 0)\n",
    "+ IF(mval_Colon_Sigmoid > 0.9, 1, 0)\n",
    "+ IF(mval_Colon_Transverse > 0.9, 1, 0)\n",
    "+ IF(mval_Esophagus_Gastroesophageal_Junction > 0.9, 1, 0)\n",
    "+ IF(mval_Esophagus_Mucosa > 0.9, 1, 0)\n",
    "+ IF(mval_Esophagus_Muscularis > 0.9, 1, 0)\n",
    "+ IF(mval_Heart_Atrial_Appendage > 0.9, 1, 0)\n",
    "+ IF(mval_Heart_Left_Ventricle > 0.9, 1, 0)\n",
    "+ IF(mval_Liver > 0.9, 1, 0)\n",
    "+ IF(mval_Lung > 0.9, 1, 0)\n",
    "+ IF(mval_Minor_Salivary_Gland > 0.9, 1, 0)\n",
    "+ IF(mval_Muscle_Skeletal > 0.9, 1, 0)\n",
    "+ IF(mval_Nerve_Tibial > 0.9, 1, 0)\n",
    "+ IF(mval_Ovary > 0.9, 1, 0)\n",
    "+ IF(mval_Pancreas > 0.9, 1, 0)\n",
    "+ IF(mval_Pituitary > 0.9, 1, 0)\n",
    "+ IF(mval_Prostate > 0.9, 1, 0)\n",
    "+ IF(mval_Skin_Not_Sun_Exposed_Suprapubic > 0.9, 1, 0)\n",
    "+ IF(mval_Skin_Sun_Exposed_Lower_leg > 0.9, 1, 0)\n",
    "+ IF(mval_Small_Intestine_Terminal_Ileum > 0.9, 1, 0)\n",
    "+ IF(mval_Spleen > 0.9, 1, 0)\n",
    "+ IF(mval_Stomach > 0.9, 1, 0)\n",
    "+ IF(mval_Testis > 0.9, 1, 0)\n",
    "+ IF(mval_Thyroid > 0.9, 1, 0)\n",
    "+ IF(mval_Uterus > 0.9, 1, 0)\n",
    "+ IF(mval_Vagina > 0.9, 1, 0)\n",
    "+ IF(mval_Whole_Blood > 0.9, 1, 0))\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Add flag for each eQTL if significant after accounting for being tested in 48 tissues\n",
    "\n",
    "```sql\n",
    "UPDATE v7\n",
    "SET sigAfterBonferroni = 0\n",
    "```\n",
    "```sql\n",
    "UPDATE v7\n",
    "SET sigAfterBonferroni = 1\n",
    "WHERE pval_nominal * 48 < pval_nominal_threshold\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get list of genes affected by eQTLs per tissue"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Bonferroni-corrected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir ../../outputFiles/GTExV7/GTExAffectedGenes\n",
    "!mkdir ../../outputFiles/GTExV7/GTExAffectedGenes/BonferroniCorrected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "for tissue in tissues:\n",
    "    affectedGenesInTissue = pd.read_sql_query(\n",
    "        'SELECT DISTINCT gene_id FROM `v7` WHERE sigAfterBonferroni = 1 AND tissue = \"' + tissue +'\"',\n",
    "        engine,\n",
    "        coerce_float=True\n",
    "    )['gene_id']\n",
    "    affectedGenesInTissue.apply(removeGeneIDVersions).to_csv('../../outputFiles/GTExV7/GTExAffectedGenes/BonferroniCorrected/'+tissue+'.txt',\n",
    "                                 index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Bonferroni-corrected - merged tissues\n",
    "\n",
    "Merge GTEx tissues such as `Adipose - Subcutaneous` and `Adipose - Visceral (Omentum)` into one tissue `Adipose` with a unique list of genes that are affected in either `Subcutaneous` or `Visceral`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir ../../outputFiles/GTExV7/GTExAffectedGenes/BonferroniCorrectedMerged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "tissuesMerged = [['Adipose', 'Adipose - Subcutaneous\", \"Adipose - Visceral (Omentum)'],\n",
    " ['Adrenal Gland', 'Adrenal Gland'],\n",
    " ['Artery', 'Artery - Aorta\", \"Artery - Coronary\", \"Artery - Tibial'],\n",
    " ['Brain', 'Brain - Amygdala\", \"Brain - Anterior cingulate cortex (BA24)\", \"Brain - Caudate (basal ganglia)\", \"Brain - Cerebellar Hemisphere\", \"Brain - Cerebellum\", \"Brain - Cortex\", \"Brain - Frontal Cortex (BA9)\", \"Brain - Hippocampus\", \"Brain - Hypothalamus\", \"Brain - Nucleus accumbens (basal ganglia)\", \"Brain - Putamen (basal ganglia)\", \"Brain - Spinal cord (cervical c-1)\", \"Brain - Substantia nigra'],\n",
    " ['Breast - Mammary Tissue', 'Breast - Mammary Tissue'],\n",
    " ['Colon', 'Colon - Sigmoid\", \"Colon - Transverse'],\n",
    " ['Esophagus', 'Esophagus - Gastroesophageal Junction\", \"Esophagus - Mucosa\", \"Esophagus - Muscularis'],\n",
    " ['Heart', 'Heart - Atrial Appendage\", \"Heart - Left Ventricle'],\n",
    " ['Liver', 'Liver'],\n",
    " ['Lung', 'Lung'],\n",
    " ['Minor Salivary Gland', 'Minor Salivary Gland'],\n",
    " ['Muscle - Skeletal', 'Muscle - Skeletal'],\n",
    " ['Nerve - Tibial', 'Nerve - Tibial'],\n",
    " ['Ovary', 'Ovary'],\n",
    " ['Pancreas', 'Pancreas'],\n",
    " ['Pituitary', 'Pituitary'],\n",
    " ['Prostate', 'Prostate'],\n",
    " ['Skin', 'Skin - Not Sun Exposed (Suprapubic)\", \"Skin - Sun Exposed (Lower leg)'],\n",
    " ['Small Intestine - Terminal Ileum', 'Small Intestine - Terminal Ileum'],\n",
    " ['Spleen', 'Spleen'],\n",
    " ['Stomach', 'Stomach'],\n",
    " ['Testis', 'Testis'],\n",
    " ['Thyroid', 'Thyroid'],\n",
    " ['Uterus', 'Uterus'],\n",
    " ['Vagina', 'Vagina'],\n",
    " ['Whole Blood', 'Whole Blood']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "for tissue in tissuesMerged:\n",
    "    affectedGenesInTissue = pd.read_sql_query(\n",
    "        'SELECT DISTINCT gene_id FROM `v7` WHERE sigAfterBonferroni = 1 AND tissue IN (\"' + tissue[1] +'\")',\n",
    "        engine,\n",
    "        coerce_float=True\n",
    "    )['gene_id']\n",
    "    affectedGenesInTissue.apply(removeGeneIDVersions).to_csv('../../outputFiles/GTExV7/GTExAffectedGenes/BonferroniCorrectedMerged/'+tissue[0]+'.txt',\n",
    "                                 index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Metasoft"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir ../../outputFiles/GTExV7/GTExAffectedGenes/Metasoft"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "for tissue in tissues:\n",
    "    affectedGenesInTissue = pd.read_sql_query(\n",
    "        'SELECT DISTINCT gene FROM `v7Metasoft` WHERE `mval_' + tissue.replace(\" - \", \"_\").replace(\" \", \"_\").replace(\"(\", \"\").replace(\")\", \"\") +'` > 0.9',\n",
    "        engine,\n",
    "        coerce_float=True\n",
    "    )['gene']\n",
    "    affectedGenesInTissue.apply(removeGeneIDVersions).to_csv('../../outputFiles/GTExV7/GTExAffectedGenes/Metasoft/'+tissue+'.txt',\n",
    "                                                             index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Count number of tissues affected by eQTL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ensembl Gene ID</th>\n",
       "      <th>Chromosome Name</th>\n",
       "      <th>Gene Start (bp)</th>\n",
       "      <th>Gene End (bp)</th>\n",
       "      <th>Strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000215405</td>\n",
       "      <td>15</td>\n",
       "      <td>20737094</td>\n",
       "      <td>20747114</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000268343</td>\n",
       "      <td>15</td>\n",
       "      <td>21004687</td>\n",
       "      <td>21005367</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000230031</td>\n",
       "      <td>15</td>\n",
       "      <td>21040701</td>\n",
       "      <td>21071643</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000138593</td>\n",
       "      <td>15</td>\n",
       "      <td>49280673</td>\n",
       "      <td>49338760</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000268531</td>\n",
       "      <td>15</td>\n",
       "      <td>22011370</td>\n",
       "      <td>22012050</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSG00000233917</td>\n",
       "      <td>15</td>\n",
       "      <td>22051853</td>\n",
       "      <td>22083227</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENSG00000166157</td>\n",
       "      <td>21</td>\n",
       "      <td>10906201</td>\n",
       "      <td>11029719</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENSG00000256715</td>\n",
       "      <td>21</td>\n",
       "      <td>14741931</td>\n",
       "      <td>14745386</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENSG00000166351</td>\n",
       "      <td>21</td>\n",
       "      <td>14982498</td>\n",
       "      <td>15013906</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSG00000269011</td>\n",
       "      <td>21</td>\n",
       "      <td>15051621</td>\n",
       "      <td>15053459</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ENSG00000168675</td>\n",
       "      <td>18</td>\n",
       "      <td>13217497</td>\n",
       "      <td>13652754</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENSG00000188992</td>\n",
       "      <td>21</td>\n",
       "      <td>15481134</td>\n",
       "      <td>15583166</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSG00000185272</td>\n",
       "      <td>21</td>\n",
       "      <td>15588451</td>\n",
       "      <td>15600693</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENSG00000182974</td>\n",
       "      <td>15</td>\n",
       "      <td>22368478</td>\n",
       "      <td>22369561</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENSG00000183706</td>\n",
       "      <td>15</td>\n",
       "      <td>22382382</td>\n",
       "      <td>22383507</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ENSG00000269182</td>\n",
       "      <td>15</td>\n",
       "      <td>22456895</td>\n",
       "      <td>22460351</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENSG00000155304</td>\n",
       "      <td>21</td>\n",
       "      <td>15743436</td>\n",
       "      <td>15755805</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ENSG00000197414</td>\n",
       "      <td>15</td>\n",
       "      <td>22736246</td>\n",
       "      <td>22746002</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ENSG00000166200</td>\n",
       "      <td>15</td>\n",
       "      <td>49398268</td>\n",
       "      <td>49447858</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ENSG00000155307</td>\n",
       "      <td>21</td>\n",
       "      <td>15857549</td>\n",
       "      <td>15955723</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ENSG00000243440</td>\n",
       "      <td>21</td>\n",
       "      <td>15964251</td>\n",
       "      <td>16031142</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ENSG00000153575</td>\n",
       "      <td>15</td>\n",
       "      <td>22833395</td>\n",
       "      <td>22873892</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ENSG00000180530</td>\n",
       "      <td>21</td>\n",
       "      <td>16333556</td>\n",
       "      <td>16437321</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ENSG00000137824</td>\n",
       "      <td>15</td>\n",
       "      <td>41028082</td>\n",
       "      <td>41048049</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ENSG00000259458</td>\n",
       "      <td>15</td>\n",
       "      <td>62929371</td>\n",
       "      <td>62937490</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ENSG00000137880</td>\n",
       "      <td>15</td>\n",
       "      <td>41056218</td>\n",
       "      <td>41059906</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ENSG00000104129</td>\n",
       "      <td>15</td>\n",
       "      <td>41060067</td>\n",
       "      <td>41099675</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ENSG00000177150</td>\n",
       "      <td>18</td>\n",
       "      <td>13663346</td>\n",
       "      <td>13726662</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ENSG00000155313</td>\n",
       "      <td>21</td>\n",
       "      <td>17102344</td>\n",
       "      <td>17252377</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ENSG00000101638</td>\n",
       "      <td>18</td>\n",
       "      <td>44259081</td>\n",
       "      <td>44339220</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20284</th>\n",
       "      <td>ENSG00000176393</td>\n",
       "      <td>1</td>\n",
       "      <td>201951500</td>\n",
       "      <td>201975275</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20285</th>\n",
       "      <td>ENSG00000163435</td>\n",
       "      <td>1</td>\n",
       "      <td>201977073</td>\n",
       "      <td>201986316</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20286</th>\n",
       "      <td>ENSG00000115464</td>\n",
       "      <td>2</td>\n",
       "      <td>61414598</td>\n",
       "      <td>61697904</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20287</th>\n",
       "      <td>ENSG00000170075</td>\n",
       "      <td>1</td>\n",
       "      <td>202091986</td>\n",
       "      <td>202102720</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20288</th>\n",
       "      <td>ENSG00000143862</td>\n",
       "      <td>1</td>\n",
       "      <td>202102532</td>\n",
       "      <td>202113869</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20289</th>\n",
       "      <td>ENSG00000143851</td>\n",
       "      <td>1</td>\n",
       "      <td>202116141</td>\n",
       "      <td>202130716</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20290</th>\n",
       "      <td>ENSG00000161714</td>\n",
       "      <td>17</td>\n",
       "      <td>43186335</td>\n",
       "      <td>43210721</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20291</th>\n",
       "      <td>ENSG00000133067</td>\n",
       "      <td>1</td>\n",
       "      <td>202163029</td>\n",
       "      <td>202288909</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20292</th>\n",
       "      <td>ENSG00000139433</td>\n",
       "      <td>12</td>\n",
       "      <td>110288748</td>\n",
       "      <td>110318293</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20293</th>\n",
       "      <td>ENSG00000077152</td>\n",
       "      <td>1</td>\n",
       "      <td>202300785</td>\n",
       "      <td>202311108</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20294</th>\n",
       "      <td>ENSG00000269897</td>\n",
       "      <td>10</td>\n",
       "      <td>22605317</td>\n",
       "      <td>22618471</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20295</th>\n",
       "      <td>ENSG00000168283</td>\n",
       "      <td>10</td>\n",
       "      <td>22610140</td>\n",
       "      <td>22620413</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20296</th>\n",
       "      <td>ENSG00000077157</td>\n",
       "      <td>1</td>\n",
       "      <td>202317827</td>\n",
       "      <td>202561834</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20297</th>\n",
       "      <td>ENSG00000139437</td>\n",
       "      <td>12</td>\n",
       "      <td>110338069</td>\n",
       "      <td>110421646</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20298</th>\n",
       "      <td>ENSG00000143858</td>\n",
       "      <td>1</td>\n",
       "      <td>202559724</td>\n",
       "      <td>202679545</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20299</th>\n",
       "      <td>ENSG00000181513</td>\n",
       "      <td>17</td>\n",
       "      <td>43209967</td>\n",
       "      <td>43221548</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20300</th>\n",
       "      <td>ENSG00000117139</td>\n",
       "      <td>1</td>\n",
       "      <td>202696526</td>\n",
       "      <td>202778598</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20301</th>\n",
       "      <td>ENSG00000184774</td>\n",
       "      <td>1</td>\n",
       "      <td>202789394</td>\n",
       "      <td>202796353</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20302</th>\n",
       "      <td>ENSG00000183155</td>\n",
       "      <td>1</td>\n",
       "      <td>202848085</td>\n",
       "      <td>202858263</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20303</th>\n",
       "      <td>ENSG00000117153</td>\n",
       "      <td>1</td>\n",
       "      <td>202860228</td>\n",
       "      <td>202897764</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20304</th>\n",
       "      <td>ENSG00000159346</td>\n",
       "      <td>1</td>\n",
       "      <td>202909951</td>\n",
       "      <td>202927700</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20305</th>\n",
       "      <td>ENSG00000159348</td>\n",
       "      <td>1</td>\n",
       "      <td>202930997</td>\n",
       "      <td>202936408</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20306</th>\n",
       "      <td>ENSG00000163444</td>\n",
       "      <td>1</td>\n",
       "      <td>202976514</td>\n",
       "      <td>202993976</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20307</th>\n",
       "      <td>ENSG00000139436</td>\n",
       "      <td>12</td>\n",
       "      <td>110367607</td>\n",
       "      <td>110434194</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20308</th>\n",
       "      <td>ENSG00000077327</td>\n",
       "      <td>10</td>\n",
       "      <td>22634399</td>\n",
       "      <td>22743153</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20309</th>\n",
       "      <td>ENSG00000186834</td>\n",
       "      <td>17</td>\n",
       "      <td>43224684</td>\n",
       "      <td>43229468</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20310</th>\n",
       "      <td>ENSG00000168517</td>\n",
       "      <td>17</td>\n",
       "      <td>43238067</td>\n",
       "      <td>43247407</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20311</th>\n",
       "      <td>ENSG00000184922</td>\n",
       "      <td>17</td>\n",
       "      <td>43298811</td>\n",
       "      <td>43324687</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20312</th>\n",
       "      <td>ENSG00000150867</td>\n",
       "      <td>10</td>\n",
       "      <td>22823778</td>\n",
       "      <td>23003484</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20313</th>\n",
       "      <td>ENSG00000082898</td>\n",
       "      <td>2</td>\n",
       "      <td>61704984</td>\n",
       "      <td>61765761</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20314 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Ensembl Gene ID Chromosome Name  Gene Start (bp)  Gene End (bp)  Strand\n",
       "0      ENSG00000215405              15         20737094       20747114      -1\n",
       "1      ENSG00000268343              15         21004687       21005367       1\n",
       "2      ENSG00000230031              15         21040701       21071643      -1\n",
       "3      ENSG00000138593              15         49280673       49338760      -1\n",
       "4      ENSG00000268531              15         22011370       22012050       1\n",
       "5      ENSG00000233917              15         22051853       22083227      -1\n",
       "6      ENSG00000166157              21         10906201       11029719      -1\n",
       "7      ENSG00000256715              21         14741931       14745386      -1\n",
       "8      ENSG00000166351              21         14982498       15013906       1\n",
       "9      ENSG00000269011              21         15051621       15053459      -1\n",
       "10     ENSG00000168675              18         13217497       13652754       1\n",
       "11     ENSG00000188992              21         15481134       15583166      -1\n",
       "12     ENSG00000185272              21         15588451       15600693       1\n",
       "13     ENSG00000182974              15         22368478       22369561       1\n",
       "14     ENSG00000183706              15         22382382       22383507       1\n",
       "15     ENSG00000269182              15         22456895       22460351       1\n",
       "16     ENSG00000155304              21         15743436       15755805      -1\n",
       "17     ENSG00000197414              15         22736246       22746002       1\n",
       "18     ENSG00000166200              15         49398268       49447858      -1\n",
       "19     ENSG00000155307              21         15857549       15955723      -1\n",
       "20     ENSG00000243440              21         15964251       16031142      -1\n",
       "21     ENSG00000153575              15         22833395       22873892       1\n",
       "22     ENSG00000180530              21         16333556       16437321      -1\n",
       "23     ENSG00000137824              15         41028082       41048049      -1\n",
       "24     ENSG00000259458              15         62929371       62937490      -1\n",
       "25     ENSG00000137880              15         41056218       41059906       1\n",
       "26     ENSG00000104129              15         41060067       41099675      -1\n",
       "27     ENSG00000177150              18         13663346       13726662      -1\n",
       "28     ENSG00000155313              21         17102344       17252377       1\n",
       "29     ENSG00000101638              18         44259081       44339220      -1\n",
       "...                ...             ...              ...            ...     ...\n",
       "20284  ENSG00000176393               1        201951500      201975275       1\n",
       "20285  ENSG00000163435               1        201977073      201986316       1\n",
       "20286  ENSG00000115464               2         61414598       61697904      -1\n",
       "20287  ENSG00000170075               1        202091986      202102720       1\n",
       "20288  ENSG00000143862               1        202102532      202113869      -1\n",
       "20289  ENSG00000143851               1        202116141      202130716      -1\n",
       "20290  ENSG00000161714              17         43186335       43210721      -1\n",
       "20291  ENSG00000133067               1        202163029      202288909       1\n",
       "20292  ENSG00000139433              12        110288748      110318293      -1\n",
       "20293  ENSG00000077152               1        202300785      202311108      -1\n",
       "20294  ENSG00000269897              10         22605317       22618471       1\n",
       "20295  ENSG00000168283              10         22610140       22620413       1\n",
       "20296  ENSG00000077157               1        202317827      202561834       1\n",
       "20297  ENSG00000139437              12        110338069      110421646       1\n",
       "20298  ENSG00000143858               1        202559724      202679545      -1\n",
       "20299  ENSG00000181513              17         43209967       43221548       1\n",
       "20300  ENSG00000117139               1        202696526      202778598      -1\n",
       "20301  ENSG00000184774               1        202789394      202796353       1\n",
       "20302  ENSG00000183155               1        202848085      202858263      -1\n",
       "20303  ENSG00000117153               1        202860228      202897764      -1\n",
       "20304  ENSG00000159346               1        202909951      202927700      -1\n",
       "20305  ENSG00000159348               1        202930997      202936408      -1\n",
       "20306  ENSG00000163444               1        202976514      202993976       1\n",
       "20307  ENSG00000139436              12        110367607      110434194      -1\n",
       "20308  ENSG00000077327              10         22634399       22743153       1\n",
       "20309  ENSG00000186834              17         43224684       43229468       1\n",
       "20310  ENSG00000168517              17         43238067       43247407       1\n",
       "20311  ENSG00000184922              17         43298811       43324687       1\n",
       "20312  ENSG00000150867              10         22823778       23003484      -1\n",
       "20313  ENSG00000082898               2         61704984       61765761      -1\n",
       "\n",
       "[20314 rows x 5 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PCGenes = pd.read_csv('../../datasets/geneLists/Ensembl/EnsV75ProteinCodingGenes1-Y.txt', sep='\\t')\n",
    "PCGenes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Bonferroni-corrected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "result = pd.DataFrame()\n",
    "for tissue in tissues:\n",
    "    tissueFrame = pd.read_csv('../../outputFiles/GTExV7/GTExAffectedGenes/BonferroniCorrected/'+tissue+'.txt',\n",
    "                              names=['Ensembl Gene ID'])\n",
    "    result = pd.concat([result, tissueFrame])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "geneTissueeQTLCounts = result['Ensembl Gene ID'].value_counts().to_frame()\n",
    "geneTissueeQTLCounts.rename(columns={'Ensembl Gene ID':'affectedTissues'}, inplace=True)\n",
    "geneTissueeQTLCounts['Ensembl Gene ID'] = geneTissueeQTLCounts.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>affectedTissues</th>\n",
       "      <th>Ensembl Gene ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000101639</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000101639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000238083</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000238083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000013573</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000013573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000254682</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000254682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000162836</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000162836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000204792</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000204792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000236682</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000236682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000214078</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000214078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000111788</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000111788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000232653</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000232653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000225138</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000225138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000234171</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000234171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000186470</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000186470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000174194</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000174194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000197932</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000197932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000160828</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000160828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000238039</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000238039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000186230</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000186230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000214401</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000214401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000144115</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000144115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000267769</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000267769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000263503</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000263503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000149231</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000149231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000128891</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000128891</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000231365</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000231365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000261186</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000261186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000113971</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000113971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000155393</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000155393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000239335</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000239335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000259065</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000259065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000162711</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000162711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000230590</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000230590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000184560</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000184560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000258534</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000258534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000261651</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000261651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000215165</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000215165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000234241</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000234241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000232901</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000232901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000225172</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000225172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000248516</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000248516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000272622</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000272622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000197591</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000197591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000254619</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000254619</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000236763</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000236763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000122787</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000122787</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000265778</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000265778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000102239</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000102239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000251164</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000251164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000241130</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000241130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000172789</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000172789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000020633</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000020633</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000231892</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000231892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000224795</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000224795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000241956</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000241956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000138483</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000138483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000256515</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000256515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000227006</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000227006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000244313</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000244313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000269821</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000269821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000141644</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000141644</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24564 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 affectedTissues  Ensembl Gene ID\n",
       "ENSG00000101639               48  ENSG00000101639\n",
       "ENSG00000238083               48  ENSG00000238083\n",
       "ENSG00000013573               48  ENSG00000013573\n",
       "ENSG00000254682               48  ENSG00000254682\n",
       "ENSG00000162836               48  ENSG00000162836\n",
       "ENSG00000204792               48  ENSG00000204792\n",
       "ENSG00000236682               48  ENSG00000236682\n",
       "ENSG00000214078               48  ENSG00000214078\n",
       "ENSG00000111788               48  ENSG00000111788\n",
       "ENSG00000232653               48  ENSG00000232653\n",
       "ENSG00000225138               48  ENSG00000225138\n",
       "ENSG00000234171               48  ENSG00000234171\n",
       "ENSG00000186470               48  ENSG00000186470\n",
       "ENSG00000174194               48  ENSG00000174194\n",
       "ENSG00000197932               48  ENSG00000197932\n",
       "ENSG00000160828               48  ENSG00000160828\n",
       "ENSG00000238039               48  ENSG00000238039\n",
       "ENSG00000186230               48  ENSG00000186230\n",
       "ENSG00000214401               48  ENSG00000214401\n",
       "ENSG00000144115               48  ENSG00000144115\n",
       "ENSG00000267769               48  ENSG00000267769\n",
       "ENSG00000263503               48  ENSG00000263503\n",
       "ENSG00000149231               48  ENSG00000149231\n",
       "ENSG00000128891               48  ENSG00000128891\n",
       "ENSG00000231365               48  ENSG00000231365\n",
       "ENSG00000261186               48  ENSG00000261186\n",
       "ENSG00000113971               48  ENSG00000113971\n",
       "ENSG00000155393               48  ENSG00000155393\n",
       "ENSG00000239335               48  ENSG00000239335\n",
       "ENSG00000259065               48  ENSG00000259065\n",
       "...                          ...              ...\n",
       "ENSG00000162711                1  ENSG00000162711\n",
       "ENSG00000230590                1  ENSG00000230590\n",
       "ENSG00000184560                1  ENSG00000184560\n",
       "ENSG00000258534                1  ENSG00000258534\n",
       "ENSG00000261651                1  ENSG00000261651\n",
       "ENSG00000215165                1  ENSG00000215165\n",
       "ENSG00000234241                1  ENSG00000234241\n",
       "ENSG00000232901                1  ENSG00000232901\n",
       "ENSG00000225172                1  ENSG00000225172\n",
       "ENSG00000248516                1  ENSG00000248516\n",
       "ENSG00000272622                1  ENSG00000272622\n",
       "ENSG00000197591                1  ENSG00000197591\n",
       "ENSG00000254619                1  ENSG00000254619\n",
       "ENSG00000236763                1  ENSG00000236763\n",
       "ENSG00000122787                1  ENSG00000122787\n",
       "ENSG00000265778                1  ENSG00000265778\n",
       "ENSG00000102239                1  ENSG00000102239\n",
       "ENSG00000251164                1  ENSG00000251164\n",
       "ENSG00000241130                1  ENSG00000241130\n",
       "ENSG00000172789                1  ENSG00000172789\n",
       "ENSG00000020633                1  ENSG00000020633\n",
       "ENSG00000231892                1  ENSG00000231892\n",
       "ENSG00000224795                1  ENSG00000224795\n",
       "ENSG00000241956                1  ENSG00000241956\n",
       "ENSG00000138483                1  ENSG00000138483\n",
       "ENSG00000256515                1  ENSG00000256515\n",
       "ENSG00000227006                1  ENSG00000227006\n",
       "ENSG00000244313                1  ENSG00000244313\n",
       "ENSG00000269821                1  ENSG00000269821\n",
       "ENSG00000141644                1  ENSG00000141644\n",
       "\n",
       "[24564 rows x 2 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "geneTissueeQTLCounts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "15738"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "affectedGenes = pd.merge(PCGenes, geneTissueeQTLCounts, how='inner', on = 'Ensembl Gene ID')\n",
    "len(affectedGenes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ensembl Gene ID</th>\n",
       "      <th>Chromosome Name</th>\n",
       "      <th>Gene Start (bp)</th>\n",
       "      <th>Gene End (bp)</th>\n",
       "      <th>Strand</th>\n",
       "      <th>affectedTissues</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000138593</td>\n",
       "      <td>15</td>\n",
       "      <td>49280673</td>\n",
       "      <td>49338760</td>\n",
       "      <td>-1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000166351</td>\n",
       "      <td>21</td>\n",
       "      <td>14982498</td>\n",
       "      <td>15013906</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000168675</td>\n",
       "      <td>18</td>\n",
       "      <td>13217497</td>\n",
       "      <td>13652754</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000188992</td>\n",
       "      <td>21</td>\n",
       "      <td>15481134</td>\n",
       "      <td>15583166</td>\n",
       "      <td>-1</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000185272</td>\n",
       "      <td>21</td>\n",
       "      <td>15588451</td>\n",
       "      <td>15600693</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Ensembl Gene ID Chromosome Name  Gene Start (bp)  Gene End (bp)  Strand  \\\n",
       "0  ENSG00000138593              15         49280673       49338760      -1   \n",
       "1  ENSG00000166351              21         14982498       15013906       1   \n",
       "2  ENSG00000168675              18         13217497       13652754       1   \n",
       "3  ENSG00000188992              21         15481134       15583166      -1   \n",
       "4  ENSG00000185272              21         15588451       15600693       1   \n",
       "\n",
       "   affectedTissues  \n",
       "0                4  \n",
       "1                1  \n",
       "2                1  \n",
       "3               13  \n",
       "4                8  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "affectedGenes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "affectedGenes.iloc[:,[0,5]].to_csv('../../outputFiles/GTExV7/genesWithNumberOfBonferroniAffectedTissues.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Bonferroni-corrected merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "result = pd.DataFrame()\n",
    "for tissue in tissuesMerged:\n",
    "    tissueFrame = pd.read_csv('../../outputFiles/GTExV7/GTExAffectedGenes/BonferroniCorrectedMerged/'+tissue[0]+'.txt',\n",
    "                              names=['Ensembl Gene ID'])\n",
    "    result = pd.concat([result, tissueFrame])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "geneTissueeQTLCounts = result['Ensembl Gene ID'].value_counts().to_frame()\n",
    "geneTissueeQTLCounts.rename(columns={'Ensembl Gene ID':'affectedTissues'}, inplace=True)\n",
    "geneTissueeQTLCounts['Ensembl Gene ID'] = geneTissueeQTLCounts.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>affectedTissues</th>\n",
       "      <th>Ensembl Gene ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000242588</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000242588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000149084</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000149084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000120539</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000120539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000263006</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000263006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000174194</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000174194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000247903</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000247903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000226124</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000226124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000111788</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000111788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000247572</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000247572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000185684</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000185684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000261015</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000261015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000229043</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000229043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000204622</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000204622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000224997</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000224997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000204520</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000204520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000166321</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000166321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000204169</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000204169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000236682</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000236682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000196756</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000196756</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000228389</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000228389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000164039</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000164039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000267769</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000267769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000182362</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000182362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000204149</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000204149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000164048</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000164048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000206149</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000206149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000261349</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000261349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000263335</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000263335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000214401</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000214401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000270081</th>\n",
       "      <td>26</td>\n",
       "      <td>ENSG00000270081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000256288</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000256288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000237396</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000237396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000234478</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000234478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000256564</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000256564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000125398</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000125398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000234025</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000234025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000258949</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000258949</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000248943</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000248943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000253307</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000253307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000115844</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000115844</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000223734</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000223734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000083307</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000083307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000235885</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000235885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000186185</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000186185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000227740</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000227740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000237882</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000237882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000204362</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000204362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000215277</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000215277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000262503</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000262503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000188827</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000188827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000236886</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000236886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000171016</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000171016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000241741</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000241741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000243716</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000243716</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000223804</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000223804</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000254764</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000254764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000185480</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000185480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000270781</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000270781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000180081</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000180081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000214027</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000214027</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24256 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 affectedTissues  Ensembl Gene ID\n",
       "ENSG00000242588               26  ENSG00000242588\n",
       "ENSG00000149084               26  ENSG00000149084\n",
       "ENSG00000120539               26  ENSG00000120539\n",
       "ENSG00000263006               26  ENSG00000263006\n",
       "ENSG00000174194               26  ENSG00000174194\n",
       "ENSG00000247903               26  ENSG00000247903\n",
       "ENSG00000226124               26  ENSG00000226124\n",
       "ENSG00000111788               26  ENSG00000111788\n",
       "ENSG00000247572               26  ENSG00000247572\n",
       "ENSG00000185684               26  ENSG00000185684\n",
       "ENSG00000261015               26  ENSG00000261015\n",
       "ENSG00000229043               26  ENSG00000229043\n",
       "ENSG00000204622               26  ENSG00000204622\n",
       "ENSG00000224997               26  ENSG00000224997\n",
       "ENSG00000204520               26  ENSG00000204520\n",
       "ENSG00000166321               26  ENSG00000166321\n",
       "ENSG00000204169               26  ENSG00000204169\n",
       "ENSG00000236682               26  ENSG00000236682\n",
       "ENSG00000196756               26  ENSG00000196756\n",
       "ENSG00000228389               26  ENSG00000228389\n",
       "ENSG00000164039               26  ENSG00000164039\n",
       "ENSG00000267769               26  ENSG00000267769\n",
       "ENSG00000182362               26  ENSG00000182362\n",
       "ENSG00000204149               26  ENSG00000204149\n",
       "ENSG00000164048               26  ENSG00000164048\n",
       "ENSG00000206149               26  ENSG00000206149\n",
       "ENSG00000261349               26  ENSG00000261349\n",
       "ENSG00000263335               26  ENSG00000263335\n",
       "ENSG00000214401               26  ENSG00000214401\n",
       "ENSG00000270081               26  ENSG00000270081\n",
       "...                          ...              ...\n",
       "ENSG00000256288                1  ENSG00000256288\n",
       "ENSG00000237396                1  ENSG00000237396\n",
       "ENSG00000234478                1  ENSG00000234478\n",
       "ENSG00000256564                1  ENSG00000256564\n",
       "ENSG00000125398                1  ENSG00000125398\n",
       "ENSG00000234025                1  ENSG00000234025\n",
       "ENSG00000258949                1  ENSG00000258949\n",
       "ENSG00000248943                1  ENSG00000248943\n",
       "ENSG00000253307                1  ENSG00000253307\n",
       "ENSG00000115844                1  ENSG00000115844\n",
       "ENSG00000223734                1  ENSG00000223734\n",
       "ENSG00000083307                1  ENSG00000083307\n",
       "ENSG00000235885                1  ENSG00000235885\n",
       "ENSG00000186185                1  ENSG00000186185\n",
       "ENSG00000227740                1  ENSG00000227740\n",
       "ENSG00000237882                1  ENSG00000237882\n",
       "ENSG00000204362                1  ENSG00000204362\n",
       "ENSG00000215277                1  ENSG00000215277\n",
       "ENSG00000262503                1  ENSG00000262503\n",
       "ENSG00000188827                1  ENSG00000188827\n",
       "ENSG00000236886                1  ENSG00000236886\n",
       "ENSG00000171016                1  ENSG00000171016\n",
       "ENSG00000241741                1  ENSG00000241741\n",
       "ENSG00000243716                1  ENSG00000243716\n",
       "ENSG00000223804                1  ENSG00000223804\n",
       "ENSG00000254764                1  ENSG00000254764\n",
       "ENSG00000185480                1  ENSG00000185480\n",
       "ENSG00000270781                1  ENSG00000270781\n",
       "ENSG00000180081                1  ENSG00000180081\n",
       "ENSG00000214027                1  ENSG00000214027\n",
       "\n",
       "[24256 rows x 2 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "geneTissueeQTLCounts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "15571"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "affectedGenes = pd.merge(PCGenes, geneTissueeQTLCounts, how='inner', on = 'Ensembl Gene ID')\n",
    "len(affectedGenes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ensembl Gene ID</th>\n",
       "      <th>Chromosome Name</th>\n",
       "      <th>Gene Start (bp)</th>\n",
       "      <th>Gene End (bp)</th>\n",
       "      <th>Strand</th>\n",
       "      <th>affectedTissues</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000138593</td>\n",
       "      <td>15</td>\n",
       "      <td>49280673</td>\n",
       "      <td>49338760</td>\n",
       "      <td>-1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000166351</td>\n",
       "      <td>21</td>\n",
       "      <td>14982498</td>\n",
       "      <td>15013906</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000168675</td>\n",
       "      <td>18</td>\n",
       "      <td>13217497</td>\n",
       "      <td>13652754</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000188992</td>\n",
       "      <td>21</td>\n",
       "      <td>15481134</td>\n",
       "      <td>15583166</td>\n",
       "      <td>-1</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000185272</td>\n",
       "      <td>21</td>\n",
       "      <td>15588451</td>\n",
       "      <td>15600693</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Ensembl Gene ID Chromosome Name  Gene Start (bp)  Gene End (bp)  Strand  \\\n",
       "0  ENSG00000138593              15         49280673       49338760      -1   \n",
       "1  ENSG00000166351              21         14982498       15013906       1   \n",
       "2  ENSG00000168675              18         13217497       13652754       1   \n",
       "3  ENSG00000188992              21         15481134       15583166      -1   \n",
       "4  ENSG00000185272              21         15588451       15600693       1   \n",
       "\n",
       "   affectedTissues  \n",
       "0                4  \n",
       "1                1  \n",
       "2                1  \n",
       "3                7  \n",
       "4                8  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "affectedGenes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "affectedGenes.iloc[:,[0,5]].to_csv('../../outputFiles/GTExV7/genesWithNumberOfBonferroniAffectedMergedTissues.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Metasoft"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "result = pd.DataFrame()\n",
    "for tissue in tissues:\n",
    "    tissueFrame = pd.read_csv('../../outputFiles/GTExV7/GTExAffectedGenes/Metasoft/'+tissue+'.txt', names=['Ensembl Gene ID'])\n",
    "    result = pd.concat([result, tissueFrame])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "geneTissueeQTLCounts = result['Ensembl Gene ID'].value_counts().to_frame()\n",
    "geneTissueeQTLCounts.rename(columns={'Ensembl Gene ID':'affectedTissues'}, inplace=True)\n",
    "geneTissueeQTLCounts['Ensembl Gene ID'] = geneTissueeQTLCounts.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>affectedTissues</th>\n",
       "      <th>Ensembl Gene ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000182670</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000182670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000136122</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000136122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000178700</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000178700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000001036</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000001036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000250132</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000250132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000271430</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000271430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000178947</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000178947</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000171311</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000171311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000105483</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000105483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000160194</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000160194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000172086</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000172086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000012174</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000012174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000197932</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000197932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000177058</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000177058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000176108</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000176108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000189045</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000189045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000180822</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000180822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000104886</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000104886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000273329</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000273329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000100058</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000100058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000102595</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000102595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000235374</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000235374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000075399</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000075399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000143458</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000143458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000141258</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000141258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000166716</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000166716</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000118961</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000118961</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000109445</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000109445</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000172890</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000172890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000213221</th>\n",
       "      <td>48</td>\n",
       "      <td>ENSG00000213221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000215104</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000215104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000241818</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000241818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000250385</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000250385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000124233</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000124233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000232930</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000232930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000228877</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000228877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000214223</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000214223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000258736</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000258736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000262999</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000262999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000259736</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000259736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000218809</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000218809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000172425</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000172425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000240086</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000240086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000271138</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000271138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000269889</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000269889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000250049</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000250049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000231883</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000231883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000182271</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000182271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000230427</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000230427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000223972</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000260871</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000260871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000227583</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000227583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000131050</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000131050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000261728</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000261728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000250129</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000250129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000233288</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000233288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000257515</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000257515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000179934</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000179934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000243290</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000243290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000204524</th>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000204524</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29056 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 affectedTissues  Ensembl Gene ID\n",
       "ENSG00000182670               48  ENSG00000182670\n",
       "ENSG00000136122               48  ENSG00000136122\n",
       "ENSG00000178700               48  ENSG00000178700\n",
       "ENSG00000001036               48  ENSG00000001036\n",
       "ENSG00000250132               48  ENSG00000250132\n",
       "ENSG00000271430               48  ENSG00000271430\n",
       "ENSG00000178947               48  ENSG00000178947\n",
       "ENSG00000171311               48  ENSG00000171311\n",
       "ENSG00000105483               48  ENSG00000105483\n",
       "ENSG00000160194               48  ENSG00000160194\n",
       "ENSG00000172086               48  ENSG00000172086\n",
       "ENSG00000012174               48  ENSG00000012174\n",
       "ENSG00000197932               48  ENSG00000197932\n",
       "ENSG00000177058               48  ENSG00000177058\n",
       "ENSG00000176108               48  ENSG00000176108\n",
       "ENSG00000189045               48  ENSG00000189045\n",
       "ENSG00000180822               48  ENSG00000180822\n",
       "ENSG00000104886               48  ENSG00000104886\n",
       "ENSG00000273329               48  ENSG00000273329\n",
       "ENSG00000100058               48  ENSG00000100058\n",
       "ENSG00000102595               48  ENSG00000102595\n",
       "ENSG00000235374               48  ENSG00000235374\n",
       "ENSG00000075399               48  ENSG00000075399\n",
       "ENSG00000143458               48  ENSG00000143458\n",
       "ENSG00000141258               48  ENSG00000141258\n",
       "ENSG00000166716               48  ENSG00000166716\n",
       "ENSG00000118961               48  ENSG00000118961\n",
       "ENSG00000109445               48  ENSG00000109445\n",
       "ENSG00000172890               48  ENSG00000172890\n",
       "ENSG00000213221               48  ENSG00000213221\n",
       "...                          ...              ...\n",
       "ENSG00000215104                1  ENSG00000215104\n",
       "ENSG00000241818                1  ENSG00000241818\n",
       "ENSG00000250385                1  ENSG00000250385\n",
       "ENSG00000124233                1  ENSG00000124233\n",
       "ENSG00000232930                1  ENSG00000232930\n",
       "ENSG00000228877                1  ENSG00000228877\n",
       "ENSG00000214223                1  ENSG00000214223\n",
       "ENSG00000258736                1  ENSG00000258736\n",
       "ENSG00000262999                1  ENSG00000262999\n",
       "ENSG00000259736                1  ENSG00000259736\n",
       "ENSG00000218809                1  ENSG00000218809\n",
       "ENSG00000172425                1  ENSG00000172425\n",
       "ENSG00000240086                1  ENSG00000240086\n",
       "ENSG00000271138                1  ENSG00000271138\n",
       "ENSG00000269889                1  ENSG00000269889\n",
       "ENSG00000250049                1  ENSG00000250049\n",
       "ENSG00000231883                1  ENSG00000231883\n",
       "ENSG00000182271                1  ENSG00000182271\n",
       "ENSG00000230427                1  ENSG00000230427\n",
       "ENSG00000223972                1  ENSG00000223972\n",
       "ENSG00000260871                1  ENSG00000260871\n",
       "ENSG00000227583                1  ENSG00000227583\n",
       "ENSG00000131050                1  ENSG00000131050\n",
       "ENSG00000261728                1  ENSG00000261728\n",
       "ENSG00000250129                1  ENSG00000250129\n",
       "ENSG00000233288                1  ENSG00000233288\n",
       "ENSG00000257515                1  ENSG00000257515\n",
       "ENSG00000179934                1  ENSG00000179934\n",
       "ENSG00000243290                1  ENSG00000243290\n",
       "ENSG00000204524                1  ENSG00000204524\n",
       "\n",
       "[29056 rows x 2 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "geneTissueeQTLCounts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "17867"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "affectedGenes = pd.merge(PCGenes, geneTissueeQTLCounts, how='inner', on = 'Ensembl Gene ID')\n",
    "len(affectedGenes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ensembl Gene ID</th>\n",
       "      <th>Chromosome Name</th>\n",
       "      <th>Gene Start (bp)</th>\n",
       "      <th>Gene End (bp)</th>\n",
       "      <th>Strand</th>\n",
       "      <th>affectedTissues</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000138593</td>\n",
       "      <td>15</td>\n",
       "      <td>49280673</td>\n",
       "      <td>49338760</td>\n",
       "      <td>-1</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000166157</td>\n",
       "      <td>21</td>\n",
       "      <td>10906201</td>\n",
       "      <td>11029719</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000168675</td>\n",
       "      <td>18</td>\n",
       "      <td>13217497</td>\n",
       "      <td>13652754</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000188992</td>\n",
       "      <td>21</td>\n",
       "      <td>15481134</td>\n",
       "      <td>15583166</td>\n",
       "      <td>-1</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000185272</td>\n",
       "      <td>21</td>\n",
       "      <td>15588451</td>\n",
       "      <td>15600693</td>\n",
       "      <td>1</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Ensembl Gene ID Chromosome Name  Gene Start (bp)  Gene End (bp)  Strand  \\\n",
       "0  ENSG00000138593              15         49280673       49338760      -1   \n",
       "1  ENSG00000166157              21         10906201       11029719      -1   \n",
       "2  ENSG00000168675              18         13217497       13652754       1   \n",
       "3  ENSG00000188992              21         15481134       15583166      -1   \n",
       "4  ENSG00000185272              21         15588451       15600693       1   \n",
       "\n",
       "   affectedTissues  \n",
       "0               24  \n",
       "1                1  \n",
       "2               10  \n",
       "3               30  \n",
       "4               48  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "affectedGenes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "affectedGenes.iloc[:,[0,5]].to_csv('../../outputFiles/GTExV7/genesWithNumberOfMetasoftAffectedTissues.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:paper-eqtls]",
   "language": "python",
   "name": "conda-env-paper-eqtls-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  },
  "toc-autonumbering": false,
  "toc-showmarkdowntxt": false,
  "toc-showtags": false
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
